// vue-tailwind-datepicker/dist/entry.5c886b32.mjs
import { openBlock as W, createElementBlock as K, createElementVNode as C, withDirectives as le, vShow as ue, toDisplayString as ne, Fragment as $e, renderList as He, inject as ce, createVNode as ee, TransitionGroup as St, withCtx as _e, normalizeClass as Me, Transition as nt, unref as N, createCommentVNode as he, withModifiers as je, cloneVNode as Dt, h as Pe, provide as de, ref as te, onMounted as Ot, watchEffect as ge, computed as fe, defineComponent as Ye, reactive as _t, nextTick as Ee, watch as Vt, isProxy as Pt, createBlock as Ie, renderSlot as Ct, mergeProps as Yt, vModelText as Tt } from "vue";
const At = { class: "flex justify-between items-center px-2 py-1.5" }, Lt = { class: "flex-shrink-0" }, Bt = { class: "inline-flex rounded-full" }, Et = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, Ft = ["d"], Nt = { class: "px-1.5 space-x-1.5 flex flex-1" }, Ht = { class: "flex-1 flex rounded-md" }, Rt = ["textContent"], It = { class: "flex-1 flex rounded-md" }, Ut = ["textContent"], zt = { class: "flex-shrink-0" }, Wt = { class: "inline-flex rounded-full" }, Zt = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, Gt = ["d"], Ue = {
  __name: "Header",
  props: {
    asPrevOrNext: Boolean,
    panel: Object,
    calendar: Object
  },
  setup(t) {
    return (a, e) => (W(), K("div", At, [
      C("div", Lt, [
        le(C("span", Bt, [
          C("button", {
            type: "button",
            class: "p-1.5 rounded-full bg-white text-vtd-secondary-600 transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: e[0] || (e[0] = (i) => t.panel.calendar ? t.calendar.onPrevious() : t.calendar.onPreviousYear())
          }, [
            (W(), K("svg", Et, [
              C("path", {
                "stroke-linecap": "round",
                "stroke-linejoin": "round",
                "stroke-width": "1.5",
                d: t.panel.calendar ? "M15 19l-7-7 7-7" : "M11 19l-7-7 7-7m8 14l-7-7 7-7"
              }, null, 8, Ft)
            ]))
          ])
        ], 512), [
          [ue, t.panel.calendar || t.panel.year]
        ])
      ]),
      C("div", Nt, [
        C("span", Ht, [
          C("button", {
            type: "button",
            class: "px-3 py-1.5 block w-full leading-relaxed rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-semibold sm:font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            textContent: ne(t.calendar.month),
            onClick: e[1] || (e[1] = (i) => t.calendar.openMonth())
          }, null, 8, Rt)
        ]),
        C("span", It, [
          C("button", {
            type: "button",
            class: "px-3 py-1.5 block w-full leading-relaxed rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-semibold sm:font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            textContent: ne(t.calendar.year),
            onClick: e[2] || (e[2] = (i) => t.calendar.openYear())
          }, null, 8, Ut)
        ])
      ]),
      C("div", zt, [
        le(C("span", Wt, [
          C("button", {
            type: "button",
            class: "p-1.5 rounded-full bg-white text-vtd-secondary-600 transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: e[3] || (e[3] = (i) => t.panel.calendar ? t.calendar.onNext() : t.calendar.onNextYear())
          }, [
            (W(), K("svg", Zt, [
              C("path", {
                "stroke-linecap": "round",
                "stroke-linejoin": "round",
                "stroke-width": "1.5",
                d: t.panel.calendar ? "M9 5l7 7-7 7" : "M13 5l7 7-7 7M5 5l7 7-7 7"
              }, null, 8, Gt)
            ]))
          ])
        ], 512), [
          [ue, t.panel.calendar || t.panel.year]
        ])
      ])
    ]));
  }
}, Kt = { class: "flex flex-wrap mt-1.5" }, Jt = { class: "flex rounded-md mt-1.5" }, qt = ["textContent", "onClick"], ze = {
  __name: "Month",
  props: {
    months: Array
  },
  emits: ["update:month"],
  setup(t, { emit: a }) {
    return (e, i) => (W(), K("div", Kt, [
      (W(!0), K($e, null, He(t.months, (n, u) => (W(), K("div", {
        key: u,
        class: "w-1/2 px-0.5"
      }, [
        C("span", Jt, [
          C("button", {
            type: "button",
            class: "px-3 py-2 block w-full leading-6 rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:hover:bg-vtd-secondary-700 dark:text-vtd-secondary-300 dark:hover:text-vtd-secondary-100 dark:focus:bg-vtd-secondary-700",
            textContent: ne(n),
            onClick: (V) => a("update:month", u)
          }, null, 8, qt)
        ])
      ]))), 128))
    ]));
  }
}, Xt = { class: "grid grid-cols-7 py-2 mt-0.5" }, Qt = ["textContent"], We = {
  __name: "Week",
  props: {
    weeks: Array
  },
  setup(t) {
    return (a, e) => (W(), K("div", Xt, [
      (W(!0), K($e, null, He(t.weeks, (i, n) => (W(), K("div", {
        key: n,
        class: "text-vtd-secondary-500 text-xs 2xl:text-sm tracking-wide font-medium text-center cursor-default dark:text-vtd-secondary-400"
      }, [
        C("span", {
          textContent: ne(i)
        }, null, 8, Qt)
      ]))), 128))
    ]));
  }
}, er = { class: "flex flex-wrap" }, tr = { class: "flex rounded-md mt-1.5" }, rr = ["textContent", "onClick"], Ze = {
  __name: "Year",
  props: {
    asPrevOrNext: Boolean,
    years: Array
  },
  emits: ["update:year"],
  setup(t, { emit: a }) {
    return (e, i) => (W(), K("div", er, [
      (W(!0), K($e, null, He(t.years, (n, u) => (W(), K("div", {
        key: u,
        class: "w-1/2 px-0.5"
      }, [
        C("span", tr, [
          C("button", {
            type: "button",
            class: "px-3 py-2 block w-full leading-6 rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:hover:bg-vtd-secondary-700 dark:text-vtd-secondary-300 dark:hover:text-vtd-secondary-100 dark:focus:bg-vtd-secondary-700",
            textContent: ne(n),
            onClick: (V) => a("update:year", n, t.asPrevOrNext)
          }, null, 8, rr)
        ])
      ]))), 128))
    ]));
  }
}, or = { class: "grid grid-cols-7 gap-y-0.5 my-1" }, ar = ["data-tooltip"], nr = ["disabled", "onClick", "onMouseenter", "onFocusin", "textContent", "data-date"], Ge = {
  __name: "Calendar",
  props: {
    asPrevOrNext: Boolean,
    calendar: Object,
    weeks: Array,
    asRange: Boolean
  },
  emits: ["update:date"],
  setup(t, { emit: a }) {
    const e = ce("isBetweenRange"), i = ce("betweenRangeClasses"), n = ce("datepickerClasses"), u = ce("atMouseOver");
    return (V, c) => (W(), K("div", or, [
      ee(St, {
        "enter-from-class": "opacity-0",
        "enter-to-class": "opacity-100",
        "enter-active-class": "transition-opacity ease-out duration-300",
        "leave-active-class": "transition-opacity ease-in duration-200",
        "leave-from-class": "opacity-100",
        "leave-to-class": "opacity-0"
      }, {
        default: _e(() => [
          (W(!0), K($e, null, He(t.calendar.date(), (f, w) => (W(), K("div", {
            key: w,
            class: Me(["relative", { "vtd-tooltip": t.asRange && f.duration() }]),
            "data-tooltip": `${f.duration()}`
          }, [
            ee(nt, {
              "enter-from-class": "opacity-0",
              "enter-to-class": "opacity-100",
              "enter-active-class": "transition-opacity ease-out duration-200",
              "leave-active-class": "transition-opacity ease-in duration-150",
              "leave-from-class": "opacity-100",
              "leave-to-class": "opacity-0"
            }, {
              default: _e(() => [
                N(e)(f) || f.hovered() ? (W(), K("span", {
                  key: 0,
                  class: Me(["absolute bg-vtd-primary-100 bg-opacity-60 dark:bg-vtd-secondary-700 dark:bg-opacity-50", N(i)(f)])
                }, null, 2)) : he("", !0)
              ]),
              _: 2
            }, 1024),
            C("button", {
              type: "button",
              class: Me(["vtd-datepicker-date relative w-[2.7rem] h-[2.7rem] lg:w-10 lg:h-10 flex justify-center items-center text-xs 2xl:text-sm focus:outline-none", [N(n)(f), t.asRange ? "transition-all" : "transition-colors"]]),
              disabled: f.disabled || f.inRange(),
              onClick: (j) => a("update:date", f, t.asPrevOrNext),
              onMouseenter: (j) => N(u)(f),
              onFocusin: (j) => N(u)(f),
              textContent: ne(f.date()),
              "data-date": f.toDate()
            }, null, 42, nr)
          ], 10, ar))), 128))
        ]),
        _: 1
      })
    ]));
  }
}, sr = {
  key: 0,
  class: "relative w-full border-t border-b-0 sm:border-t-0 sm:border-b lg:border-b-0 lg:border-r border-black/[.1] order-last sm:order-none dark:border-vtd-secondary-700/[1] sm:mt-1 lg:mr-1 sm:mb-1 lg:mb-0 sm:mx-1 lg:mx-0"
}, lr = {
  key: 0,
  class: "grid grid-cols-2 sm:grid-cols-3 gap-1 lg:block w-full pr-5 sm:pr-6 mt-1.5 sm:mt-0 sm:mb-1.5 lg:mb-0"
}, ir = ["onClick", "textContent"], ur = {
  key: 1,
  class: "grid grid-cols-2 sm:grid-cols-3 gap-1 lg:block w-full pr-5 sm:pr-6 mt-1.5 sm:mt-0 sm:mb-1.5 lg:mb-0"
}, ot = {
  __name: "Shortcut",
  props: {
    shortcuts: [Boolean, Function],
    close: Function,
    asRange: Boolean,
    asSingle: Boolean,
    i18n: Object
  },
  setup(t) {
    const a = t, e = ce("setToToday"), i = ce("setToYesterday"), n = ce("setToLastDay"), u = ce("setToThisMonth"), V = ce("setToLastMonth"), c = ce("setToCustomShortcut"), f = () => typeof a.shortcuts == "function" ? a.shortcuts() : !1;
    return (w, j) => a.asRange && a.asSingle || a.asRange && !a.asSingle ? (W(), K("div", sr, [
      f() ? (W(), K("ol", lr, [
        (W(!0), K($e, null, He(f(), (y, $) => (W(), K("li", { key: $ }, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: je((L) => N(c)(y, t.close), ["prevent"]),
            textContent: ne(y.label)
          }, null, 8, ir)
        ]))), 128))
      ])) : (W(), K("ol", ur, [
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: j[0] || (j[0] = je((y) => N(e)(t.close), ["prevent"]))
          }, ne(a.i18n.today), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: j[1] || (j[1] = je((y) => N(i)(t.close), ["prevent"]))
          }, ne(a.i18n.yesterday), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: j[2] || (j[2] = je((y) => N(n)(7, t.close), ["prevent"]))
          }, ne(a.i18n.past(7)), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: j[3] || (j[3] = je((y) => N(n)(30, t.close), ["prevent"]))
          }, ne(a.i18n.past(30)), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: j[4] || (j[4] = je((y) => N(u)(t.close), ["prevent"]))
          }, ne(a.i18n.currentMonth), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: j[5] || (j[5] = je((y) => N(V)(t.close), ["prevent"]))
          }, ne(a.i18n.pastMonth), 1)
        ])
      ]))
    ])) : he("", !0);
  }
};
function we(t, a, ...e) {
  if (t in a) {
    let n = a[t];
    return typeof n == "function" ? n(...e) : n;
  }
  let i = new Error(`Tried to handle "${t}" but there is no handler defined. Only defined handlers are: ${Object.keys(a).map((n) => `"${n}"`).join(", ")}.`);
  throw Error.captureStackTrace && Error.captureStackTrace(i, we), i;
}
var Fe = ((t) => (t[t.None = 0] = "None", t[t.RenderStrategy = 1] = "RenderStrategy", t[t.Static = 2] = "Static", t))(Fe || {}), dr = ((t) => (t[t.Unmount = 0] = "Unmount", t[t.Hidden = 1] = "Hidden", t))(dr || {});
function Te({ visible: t = !0, features: a = 0, ourProps: e, theirProps: i, ...n }) {
  var u;
  let V = cr(i, e), c = Object.assign(n, { props: V });
  if (t || a & 2 && V.static)
    return Xe(c);
  if (a & 1) {
    let f = (u = V.unmount) == null || u ? 0 : 1;
    return we(f, { [0]() {
      return null;
    }, [1]() {
      return Xe({ ...n, props: { ...V, hidden: !0, style: { display: "none" } } });
    } });
  }
  return Xe(c);
}
function Xe({ props: t, attrs: a, slots: e, slot: i, name: n }) {
  var u;
  let { as: V, ...c } = lt(t, ["unmount", "static"]), f = (u = e.default) == null ? void 0 : u.call(e, i), w = {};
  if (i) {
    let j = !1, y = [];
    for (let [$, L] of Object.entries(i))
      typeof L == "boolean" && (j = !0), L === !0 && y.push($);
    j && (w["data-headlessui-state"] = y.join(" "));
  }
  if (V === "template") {
    if (f = st(f), Object.keys(c).length > 0 || Object.keys(a).length > 0) {
      let [j, ...y] = f != null ? f : [];
      if (!fr(j) || y.length > 0)
        throw new Error(['Passing props on "template"!', "", `The current component <${n} /> is rendering a "template".`, "However we need to passthrough the following props:", Object.keys(c).concat(Object.keys(a)).sort(($, L) => $.localeCompare(L)).map(($) => `  - ${$}`).join(`
`), "", "You can apply a few solutions:", ['Add an `as="..."` prop, to ensure that we render an actual element instead of a "template".', "Render a single element as the child so that we can forward the props onto that element."].map(($) => `  - ${$}`).join(`
`)].join(`
`));
      return Dt(j, Object.assign({}, c, w));
    }
    return Array.isArray(f) && f.length === 1 ? f[0] : f;
  }
  return Pe(V, Object.assign({}, c, w), f);
}
function st(t) {
  return t.flatMap((a) => a.type === $e ? st(a.children) : [a]);
}
function cr(...t) {
  if (t.length === 0)
    return {};
  if (t.length === 1)
    return t[0];
  let a = {}, e = {};
  for (let i of t)
    for (let n in i)
      n.startsWith("on") && typeof i[n] == "function" ? (e[n] != null || (e[n] = []), e[n].push(i[n])) : a[n] = i[n];
  if (a.disabled || a["aria-disabled"])
    return Object.assign(a, Object.fromEntries(Object.keys(e).map((i) => [i, void 0])));
  for (let i in e)
    Object.assign(a, { [i](n, ...u) {
      let V = e[i];
      for (let c of V) {
        if (n instanceof Event && n.defaultPrevented)
          return;
        c(n, ...u);
      }
    } });
  return a;
}
function lt(t, a = []) {
  let e = Object.assign({}, t);
  for (let i of a)
    i in e && delete e[i];
  return e;
}
function fr(t) {
  return t == null ? !1 : typeof t.type == "string" || typeof t.type == "object" || typeof t.type == "function";
}
let mr = 0;
function pr() {
  return ++mr;
}
function Ce() {
  return pr();
}
var ke = ((t) => (t.Space = " ", t.Enter = "Enter", t.Escape = "Escape", t.Backspace = "Backspace", t.Delete = "Delete", t.ArrowLeft = "ArrowLeft", t.ArrowUp = "ArrowUp", t.ArrowRight = "ArrowRight", t.ArrowDown = "ArrowDown", t.Home = "Home", t.End = "End", t.PageUp = "PageUp", t.PageDown = "PageDown", t.Tab = "Tab", t))(ke || {});
function U(t) {
  var a;
  return t == null || t.value == null ? null : (a = t.value.$el) != null ? a : t.value;
}
let it = Symbol("Context");
var Ne = ((t) => (t[t.Open = 0] = "Open", t[t.Closed = 1] = "Closed", t))(Ne || {});
function ut() {
  return ce(it, null);
}
function vr(t) {
  de(it, t);
}
function at(t, a) {
  if (t)
    return t;
  let e = a != null ? a : "button";
  if (typeof e == "string" && e.toLowerCase() === "button")
    return "button";
}
function hr(t, a) {
  let e = te(at(t.value.type, t.value.as));
  return Ot(() => {
    e.value = at(t.value.type, t.value.as);
  }), ge(() => {
    var i;
    e.value || !U(a) || U(a) instanceof HTMLButtonElement && !((i = U(a)) != null && i.hasAttribute("type")) && (e.value = "button");
  }), e;
}
const Je = typeof window > "u" || typeof document > "u";
function Re(t) {
  if (Je)
    return null;
  if (t instanceof Node)
    return t.ownerDocument;
  if (t != null && t.hasOwnProperty("value")) {
    let a = U(t);
    if (a)
      return a.ownerDocument;
  }
  return document;
}
let et = ["[contentEditable=true]", "[tabindex]", "a[href]", "area[href]", "button:not([disabled])", "iframe", "input:not([disabled])", "select:not([disabled])", "textarea:not([disabled])"].map((t) => `${t}:not([tabindex='-1'])`).join(",");
var Se = ((t) => (t[t.First = 1] = "First", t[t.Previous = 2] = "Previous", t[t.Next = 4] = "Next", t[t.Last = 8] = "Last", t[t.WrapAround = 16] = "WrapAround", t[t.NoScroll = 32] = "NoScroll", t))(Se || {}), yr = ((t) => (t[t.Error = 0] = "Error", t[t.Overflow = 1] = "Overflow", t[t.Success = 2] = "Success", t[t.Underflow = 3] = "Underflow", t))(yr || {}), br = ((t) => (t[t.Previous = -1] = "Previous", t[t.Next = 1] = "Next", t))(br || {});
function dt(t = document.body) {
  return t == null ? [] : Array.from(t.querySelectorAll(et));
}
var rt = ((t) => (t[t.Strict = 0] = "Strict", t[t.Loose = 1] = "Loose", t))(rt || {});
function ct(t, a = 0) {
  var e;
  return t === ((e = Re(t)) == null ? void 0 : e.body) ? !1 : we(a, { [0]() {
    return t.matches(et);
  }, [1]() {
    let i = t;
    for (; i !== null; ) {
      if (i.matches(et))
        return !0;
      i = i.parentElement;
    }
    return !1;
  } });
}
let xr = ["textarea", "input"].join(",");
function gr(t) {
  var a, e;
  return (e = (a = t == null ? void 0 : t.matches) == null ? void 0 : a.call(t, xr)) != null ? e : !1;
}
function kr(t, a = (e) => e) {
  return t.slice().sort((e, i) => {
    let n = a(e), u = a(i);
    if (n === null || u === null)
      return 0;
    let V = n.compareDocumentPosition(u);
    return V & Node.DOCUMENT_POSITION_FOLLOWING ? -1 : V & Node.DOCUMENT_POSITION_PRECEDING ? 1 : 0;
  });
}
function Ve(t, a, e = !0, i = null) {
  var n;
  let u = (n = Array.isArray(t) ? t.length > 0 ? t[0].ownerDocument : document : t == null ? void 0 : t.ownerDocument) != null ? n : document, V = Array.isArray(t) ? e ? kr(t) : t : dt(t);
  i = i != null ? i : u.activeElement;
  let c = (() => {
    if (a & 5)
      return 1;
    if (a & 10)
      return -1;
    throw new Error("Missing Focus.First, Focus.Previous, Focus.Next or Focus.Last");
  })(), f = (() => {
    if (a & 1)
      return 0;
    if (a & 2)
      return Math.max(0, V.indexOf(i)) - 1;
    if (a & 4)
      return Math.max(0, V.indexOf(i)) + 1;
    if (a & 8)
      return V.length - 1;
    throw new Error("Missing Focus.First, Focus.Previous, Focus.Next or Focus.Last");
  })(), w = a & 32 ? { preventScroll: !0 } : {}, j = 0, y = V.length, $;
  do {
    if (j >= y || j + y <= 0)
      return 0;
    let L = f + j;
    if (a & 16)
      L = (L + y) % y;
    else {
      if (L < 0)
        return 3;
      if (L >= y)
        return 1;
    }
    $ = V[L], $ == null || $.focus(w), j += c;
  } while ($ !== u.activeElement);
  return a & 6 && gr($) && $.select(), $.hasAttribute("tabindex") || $.setAttribute("tabindex", "0"), 2;
}
function Qe(t, a, e) {
  Je || ge((i) => {
    document.addEventListener(t, a, e), i(() => document.removeEventListener(t, a, e));
  });
}
function wr(t, a, e = fe(() => !0)) {
  function i(u, V) {
    if (!e.value || u.defaultPrevented)
      return;
    let c = V(u);
    if (c === null || !c.ownerDocument.documentElement.contains(c))
      return;
    let f = function w(j) {
      return typeof j == "function" ? w(j()) : Array.isArray(j) || j instanceof Set ? j : [j];
    }(t);
    for (let w of f) {
      if (w === null)
        continue;
      let j = w instanceof HTMLElement ? w : U(w);
      if (j != null && j.contains(c))
        return;
    }
    return !ct(c, rt.Loose) && c.tabIndex !== -1 && u.preventDefault(), a(u, c);
  }
  let n = te(null);
  Qe("mousedown", (u) => {
    e.value && (n.value = u.target);
  }, !0), Qe("click", (u) => {
    !n.value || (i(u, () => n.value), n.value = null);
  }, !0), Qe("blur", (u) => i(u, () => window.document.activeElement instanceof HTMLIFrameElement ? window.document.activeElement : null), !0);
}
var Ke = ((t) => (t[t.None = 1] = "None", t[t.Focusable = 2] = "Focusable", t[t.Hidden = 4] = "Hidden", t))(Ke || {});
let tt = Ye({ name: "Hidden", props: { as: { type: [Object, String], default: "div" }, features: { type: Number, default: 1 } }, setup(t, { slots: a, attrs: e }) {
  return () => {
    let { features: i, ...n } = t, u = { "aria-hidden": (i & 2) === 2 ? !0 : void 0, style: { position: "fixed", top: 1, left: 1, width: 1, height: 0, padding: 0, margin: -1, overflow: "hidden", clip: "rect(0, 0, 0, 0)", whiteSpace: "nowrap", borderWidth: "0", ...(i & 4) === 4 && (i & 2) !== 2 && { display: "none" } } };
    return Te({ ourProps: u, theirProps: n, slot: {}, attrs: e, slots: a, name: "Hidden" });
  };
} });
function $r(t, a, e) {
  Je || ge((i) => {
    window.addEventListener(t, a, e), i(() => window.removeEventListener(t, a, e));
  });
}
var De = ((t) => (t[t.Forwards = 0] = "Forwards", t[t.Backwards = 1] = "Backwards", t))(De || {});
function ft() {
  let t = te(0);
  return $r("keydown", (a) => {
    a.key === "Tab" && (t.value = a.shiftKey ? 1 : 0);
  }), t;
}
function jr(t, a, e, i) {
  Je || ge((n) => {
    t = t != null ? t : window, t.addEventListener(a, e, i), n(() => t.removeEventListener(a, e, i));
  });
}
var Mr = ((t) => (t[t.Open = 0] = "Open", t[t.Closed = 1] = "Closed", t))(Mr || {});
let mt = Symbol("PopoverContext");
function qe(t) {
  let a = ce(mt, null);
  if (a === null) {
    let e = new Error(`<${t} /> is missing a parent <${yt.name} /> component.`);
    throw Error.captureStackTrace && Error.captureStackTrace(e, qe), e;
  }
  return a;
}
let pt = Symbol("PopoverGroupContext");
function vt() {
  return ce(pt, null);
}
let ht = Symbol("PopoverPanelContext");
function Sr() {
  return ce(ht, null);
}
let yt = Ye({ name: "Popover", props: { as: { type: [Object, String], default: "div" } }, setup(t, { slots: a, attrs: e, expose: i }) {
  var n;
  let u = `headlessui-popover-button-${Ce()}`, V = `headlessui-popover-panel-${Ce()}`, c = te(null);
  i({ el: c, $el: c });
  let f = te(1), w = te(null), j = te(null), y = te(null), $ = te(null), L = fe(() => Re(c)), G = fe(() => {
    if (!U(w) || !U($))
      return !1;
    for (let B of document.querySelectorAll("body > *"))
      if (Number(B == null ? void 0 : B.contains(U(w))) ^ Number(B == null ? void 0 : B.contains(U($))))
        return !0;
    return !1;
  }), D = { popoverState: f, buttonId: u, panelId: V, panel: $, button: w, isPortalled: G, beforePanelSentinel: j, afterPanelSentinel: y, togglePopover() {
    f.value = we(f.value, { [0]: 1, [1]: 0 });
  }, closePopover() {
    f.value !== 1 && (f.value = 1);
  }, close(B) {
    D.closePopover();
    let l = (() => B ? B instanceof HTMLElement ? B : B.value instanceof HTMLElement ? U(B) : U(D.button) : U(D.button))();
    l == null || l.focus();
  } };
  de(mt, D), vr(fe(() => we(f.value, { [0]: Ne.Open, [1]: Ne.Closed })));
  let b = { buttonId: u, panelId: V, close() {
    D.closePopover();
  } }, v = vt(), g = v == null ? void 0 : v.registerPopover;
  function T() {
    var B, l, r, M;
    return (M = v == null ? void 0 : v.isFocusWithinPopoverGroup()) != null ? M : ((B = L.value) == null ? void 0 : B.activeElement) && (((l = U(w)) == null ? void 0 : l.contains(L.value.activeElement)) || ((r = U($)) == null ? void 0 : r.contains(L.value.activeElement)));
  }
  return ge(() => g == null ? void 0 : g(b)), jr((n = L.value) == null ? void 0 : n.defaultView, "focus", (B) => {
    var l, r;
    f.value === 0 && (T() || !w || !$ || (l = U(D.beforePanelSentinel)) != null && l.contains(B.target) || (r = U(D.afterPanelSentinel)) != null && r.contains(B.target) || D.closePopover());
  }, !0), wr([w, $], (B, l) => {
    var r;
    D.closePopover(), ct(l, rt.Loose) || (B.preventDefault(), (r = U(w)) == null || r.focus());
  }, fe(() => f.value === 0)), () => {
    let B = { open: f.value === 0, close: D.close };
    return Te({ theirProps: t, ourProps: { ref: c }, slot: B, slots: a, attrs: e, name: "Popover" });
  };
} }), Dr = Ye({ name: "PopoverButton", props: { as: { type: [Object, String], default: "button" }, disabled: { type: [Boolean], default: !1 } }, inheritAttrs: !1, setup(t, { attrs: a, slots: e, expose: i }) {
  let n = qe("PopoverButton"), u = fe(() => Re(n.button));
  i({ el: n.button, $el: n.button });
  let V = vt(), c = V == null ? void 0 : V.closeOthers, f = Sr(), w = f === null ? !1 : f === n.panelId, j = te(null), y = `headlessui-focus-sentinel-${Ce()}`;
  w || ge(() => {
    n.button.value = j.value;
  });
  let $ = hr(fe(() => ({ as: t.as, type: a.type })), j);
  function L(v) {
    var g, T, B, l, r;
    if (w) {
      if (n.popoverState.value === 1)
        return;
      switch (v.key) {
        case ke.Space:
        case ke.Enter:
          v.preventDefault(), (T = (g = v.target).click) == null || T.call(g), n.closePopover(), (B = U(n.button)) == null || B.focus();
          break;
      }
    } else
      switch (v.key) {
        case ke.Space:
        case ke.Enter:
          v.preventDefault(), v.stopPropagation(), n.popoverState.value === 1 && (c == null || c(n.buttonId)), n.togglePopover();
          break;
        case ke.Escape:
          if (n.popoverState.value !== 0)
            return c == null ? void 0 : c(n.buttonId);
          if (!U(n.button) || ((l = u.value) == null ? void 0 : l.activeElement) && !((r = U(n.button)) != null && r.contains(u.value.activeElement)))
            return;
          v.preventDefault(), v.stopPropagation(), n.closePopover();
          break;
      }
  }
  function G(v) {
    w || v.key === ke.Space && v.preventDefault();
  }
  function D(v) {
    var g, T;
    t.disabled || (w ? (n.closePopover(), (g = U(n.button)) == null || g.focus()) : (v.preventDefault(), v.stopPropagation(), n.popoverState.value === 1 && (c == null || c(n.buttonId)), n.togglePopover(), (T = U(n.button)) == null || T.focus()));
  }
  function b(v) {
    v.preventDefault(), v.stopPropagation();
  }
  return () => {
    let v = n.popoverState.value === 0, g = { open: v }, T = w ? { ref: j, type: $.value, onKeydown: L, onClick: D } : { ref: j, id: n.buttonId, type: $.value, "aria-expanded": t.disabled ? void 0 : n.popoverState.value === 0, "aria-controls": U(n.panel) ? n.panelId : void 0, disabled: t.disabled ? !0 : void 0, onKeydown: L, onKeyup: G, onClick: D, onMousedown: b }, B = ft();
    function l() {
      let r = U(n.panel);
      if (!r)
        return;
      function M() {
        we(B.value, { [De.Forwards]: () => Ve(r, Se.First), [De.Backwards]: () => Ve(r, Se.Last) });
      }
      M();
    }
    return Pe($e, [Te({ ourProps: T, theirProps: { ...a, ...t }, slot: g, attrs: a, slots: e, name: "PopoverButton" }), v && !w && n.isPortalled.value && Pe(tt, { id: y, features: Ke.Focusable, as: "button", type: "button", onFocus: l })]);
  };
} }), Or = Ye({ name: "PopoverOverlay", props: { as: { type: [Object, String], default: "div" }, static: { type: Boolean, default: !1 }, unmount: { type: Boolean, default: !0 } }, setup(t, { attrs: a, slots: e }) {
  let i = qe("PopoverOverlay"), n = `headlessui-popover-overlay-${Ce()}`, u = ut(), V = fe(() => u !== null ? u.value === Ne.Open : i.popoverState.value === 0);
  function c() {
    i.closePopover();
  }
  return () => {
    let f = { open: i.popoverState.value === 0 };
    return Te({ ourProps: { id: n, "aria-hidden": !0, onClick: c }, theirProps: t, slot: f, attrs: a, slots: e, features: Fe.RenderStrategy | Fe.Static, visible: V.value, name: "PopoverOverlay" });
  };
} }), _r = Ye({ name: "PopoverPanel", props: { as: { type: [Object, String], default: "div" }, static: { type: Boolean, default: !1 }, unmount: { type: Boolean, default: !0 }, focus: { type: Boolean, default: !1 } }, inheritAttrs: !1, setup(t, { attrs: a, slots: e, expose: i }) {
  let { focus: n } = t, u = qe("PopoverPanel"), V = fe(() => Re(u.panel)), c = `headlessui-focus-sentinel-before-${Ce()}`, f = `headlessui-focus-sentinel-after-${Ce()}`;
  i({ el: u.panel, $el: u.panel }), de(ht, u.panelId), ge(() => {
    var b, v;
    if (!n || u.popoverState.value !== 0 || !u.panel)
      return;
    let g = (b = V.value) == null ? void 0 : b.activeElement;
    (v = U(u.panel)) != null && v.contains(g) || Ve(U(u.panel), Se.First);
  });
  let w = ut(), j = fe(() => w !== null ? w.value === Ne.Open : u.popoverState.value === 0);
  function y(b) {
    var v, g;
    switch (b.key) {
      case ke.Escape:
        if (u.popoverState.value !== 0 || !U(u.panel) || V.value && !((v = U(u.panel)) != null && v.contains(V.value.activeElement)))
          return;
        b.preventDefault(), b.stopPropagation(), u.closePopover(), (g = U(u.button)) == null || g.focus();
        break;
    }
  }
  function $(b) {
    var v, g, T, B, l;
    let r = b.relatedTarget;
    !r || !U(u.panel) || (v = U(u.panel)) != null && v.contains(r) || (u.closePopover(), (((T = (g = U(u.beforePanelSentinel)) == null ? void 0 : g.contains) == null ? void 0 : T.call(g, r)) || ((l = (B = U(u.afterPanelSentinel)) == null ? void 0 : B.contains) == null ? void 0 : l.call(B, r))) && r.focus({ preventScroll: !0 }));
  }
  let L = ft();
  function G() {
    let b = U(u.panel);
    if (!b)
      return;
    function v() {
      we(L.value, { [De.Forwards]: () => {
        Ve(b, Se.Next);
      }, [De.Backwards]: () => {
        var g;
        (g = U(u.button)) == null || g.focus({ preventScroll: !0 });
      } });
    }
    v();
  }
  function D() {
    let b = U(u.panel);
    if (!b)
      return;
    function v() {
      we(L.value, { [De.Forwards]: () => {
        var g, T;
        let B = U(u.button), l = U(u.panel);
        if (!B)
          return;
        let r = dt(), M = r.indexOf(B), H = r.slice(0, M + 1), _ = [...r.slice(M + 1), ...H];
        for (let P of _.slice())
          if (((T = (g = P == null ? void 0 : P.id) == null ? void 0 : g.startsWith) == null ? void 0 : T.call(g, "headlessui-focus-sentinel-")) || (l == null ? void 0 : l.contains(P))) {
            let Z = _.indexOf(P);
            Z !== -1 && _.splice(Z, 1);
          }
        Ve(_, Se.First, !1);
      }, [De.Backwards]: () => Ve(b, Se.Previous) });
    }
    v();
  }
  return () => {
    let b = { open: u.popoverState.value === 0, close: u.close }, v = { ref: u.panel, id: u.panelId, onKeydown: y, onFocusout: n && u.popoverState.value === 0 ? $ : void 0, tabIndex: -1 };
    return Te({ ourProps: v, theirProps: { ...a, ...lt(t, ["focus"]) }, attrs: a, slot: b, slots: { ...e, default: (...g) => {
      var T;
      return [Pe($e, [j.value && u.isPortalled.value && Pe(tt, { id: c, ref: u.beforePanelSentinel, features: Ke.Focusable, as: "button", type: "button", onFocus: G }), (T = e.default) == null ? void 0 : T.call(e, ...g), j.value && u.isPortalled.value && Pe(tt, { id: f, ref: u.afterPanelSentinel, features: Ke.Focusable, as: "button", type: "button", onFocus: D })])];
    } }, features: Fe.RenderStrategy | Fe.Static, visible: j.value, name: "PopoverPanel" });
  };
} });
Ye({ name: "PopoverGroup", props: { as: { type: [Object, String], default: "div" } }, setup(t, { attrs: a, slots: e, expose: i }) {
  let n = te(null), u = te([]), V = fe(() => Re(n));
  i({ el: n, $el: n });
  function c(y) {
    let $ = u.value.indexOf(y);
    $ !== -1 && u.value.splice($, 1);
  }
  function f(y) {
    return u.value.push(y), () => {
      c(y);
    };
  }
  function w() {
    var y;
    let $ = V.value;
    if (!$)
      return !1;
    let L = $.activeElement;
    return (y = U(n)) != null && y.contains(L) ? !0 : u.value.some((G) => {
      var D, b;
      return ((D = $.getElementById(G.buttonId)) == null ? void 0 : D.contains(L)) || ((b = $.getElementById(G.panelId)) == null ? void 0 : b.contains(L));
    });
  }
  function j(y) {
    for (let $ of u.value)
      $.buttonId !== y && $.close();
  }
  return de(pt, { registerPopover: f, unregisterPopover: c, isFocusWithinPopoverGroup: w, closeOthers: j }), () => Te({ ourProps: { ref: n }, theirProps: t, slot: {}, attrs: a, slots: e, name: "PopoverGroup" });
} });
var Oe = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, bt = { exports: {} };
(function(t, a) {
  (function(e, i) {
    t.exports = i();
  })(Oe, function() {
    var e = 1e3, i = 6e4, n = 36e5, u = "millisecond", V = "second", c = "minute", f = "hour", w = "day", j = "week", y = "month", $ = "quarter", L = "year", G = "date", D = "Invalid Date", b = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, v = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, g = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(O) {
      var x = ["th", "st", "nd", "rd"], m = O % 100;
      return "[" + O + (x[(m - 20) % 10] || x[m] || x[0]) + "]";
    } }, T = function(O, x, m) {
      var E = String(O);
      return !E || E.length >= x ? O : "" + Array(x + 1 - E.length).join(m) + O;
    }, B = { s: T, z: function(O) {
      var x = -O.utcOffset(), m = Math.abs(x), E = Math.floor(m / 60), S = m % 60;
      return (x <= 0 ? "+" : "-") + T(E, 2, "0") + ":" + T(S, 2, "0");
    }, m: function O(x, m) {
      if (x.date() < m.date())
        return -O(m, x);
      var E = 12 * (m.year() - x.year()) + (m.month() - x.month()), S = x.clone().add(E, y), F = m - S < 0, A = x.clone().add(E + (F ? -1 : 1), y);
      return +(-(E + (m - S) / (F ? S - A : A - S)) || 0);
    }, a: function(O) {
      return O < 0 ? Math.ceil(O) || 0 : Math.floor(O);
    }, p: function(O) {
      return { M: y, y: L, w: j, d: w, D: G, h: f, m: c, s: V, ms: u, Q: $ }[O] || String(O || "").toLowerCase().replace(/s$/, "");
    }, u: function(O) {
      return O === void 0;
    } }, l = "en", r = {};
    r[l] = g;
    var M = function(O) {
      return O instanceof Z;
    }, H = function O(x, m, E) {
      var S;
      if (!x)
        return l;
      if (typeof x == "string") {
        var F = x.toLowerCase();
        r[F] && (S = F), m && (r[F] = m, S = F);
        var A = x.split("-");
        if (!S && A.length > 1)
          return O(A[0]);
      } else {
        var R = x.name;
        r[R] = x, S = R;
      }
      return !E && S && (l = S), S || !E && l;
    }, _ = function(O, x) {
      if (M(O))
        return O.clone();
      var m = typeof x == "object" ? x : {};
      return m.date = O, m.args = arguments, new Z(m);
    }, P = B;
    P.l = H, P.i = M, P.w = function(O, x) {
      return _(O, { locale: x.$L, utc: x.$u, x: x.$x, $offset: x.$offset });
    };
    var Z = function() {
      function O(m) {
        this.$L = H(m.locale, null, !0), this.parse(m);
      }
      var x = O.prototype;
      return x.parse = function(m) {
        this.$d = function(E) {
          var S = E.date, F = E.utc;
          if (S === null)
            return new Date(NaN);
          if (P.u(S))
            return new Date();
          if (S instanceof Date)
            return new Date(S);
          if (typeof S == "string" && !/Z$/i.test(S)) {
            var A = S.match(b);
            if (A) {
              var R = A[2] - 1 || 0, q = (A[7] || "0").substring(0, 3);
              return F ? new Date(Date.UTC(A[1], R, A[3] || 1, A[4] || 0, A[5] || 0, A[6] || 0, q)) : new Date(A[1], R, A[3] || 1, A[4] || 0, A[5] || 0, A[6] || 0, q);
            }
          }
          return new Date(S);
        }(m), this.$x = m.x || {}, this.init();
      }, x.init = function() {
        var m = this.$d;
        this.$y = m.getFullYear(), this.$M = m.getMonth(), this.$D = m.getDate(), this.$W = m.getDay(), this.$H = m.getHours(), this.$m = m.getMinutes(), this.$s = m.getSeconds(), this.$ms = m.getMilliseconds();
      }, x.$utils = function() {
        return P;
      }, x.isValid = function() {
        return this.$d.toString() !== D;
      }, x.isSame = function(m, E) {
        var S = _(m);
        return this.startOf(E) <= S && S <= this.endOf(E);
      }, x.isAfter = function(m, E) {
        return _(m) < this.startOf(E);
      }, x.isBefore = function(m, E) {
        return this.endOf(E) < _(m);
      }, x.$g = function(m, E, S) {
        return P.u(m) ? this[E] : this.set(S, m);
      }, x.unix = function() {
        return Math.floor(this.valueOf() / 1e3);
      }, x.valueOf = function() {
        return this.$d.getTime();
      }, x.startOf = function(m, E) {
        var S = this, F = !!P.u(E) || E, A = P.p(m), R = function(ve, oe) {
          var ie = P.w(S.$u ? Date.UTC(S.$y, oe, ve) : new Date(S.$y, oe, ve), S);
          return F ? ie : ie.endOf(w);
        }, q = function(ve, oe) {
          return P.w(S.toDate()[ve].apply(S.toDate("s"), (F ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(oe)), S);
        }, X = this.$W, re = this.$M, pe = this.$D, me = "set" + (this.$u ? "UTC" : "");
        switch (A) {
          case L:
            return F ? R(1, 0) : R(31, 11);
          case y:
            return F ? R(1, re) : R(0, re + 1);
          case j:
            var xe = this.$locale().weekStart || 0, se = (X < xe ? X + 7 : X) - xe;
            return R(F ? pe - se : pe + (6 - se), re);
          case w:
          case G:
            return q(me + "Hours", 0);
          case f:
            return q(me + "Minutes", 1);
          case c:
            return q(me + "Seconds", 2);
          case V:
            return q(me + "Milliseconds", 3);
          default:
            return this.clone();
        }
      }, x.endOf = function(m) {
        return this.startOf(m, !1);
      }, x.$set = function(m, E) {
        var S, F = P.p(m), A = "set" + (this.$u ? "UTC" : ""), R = (S = {}, S[w] = A + "Date", S[G] = A + "Date", S[y] = A + "Month", S[L] = A + "FullYear", S[f] = A + "Hours", S[c] = A + "Minutes", S[V] = A + "Seconds", S[u] = A + "Milliseconds", S)[F], q = F === w ? this.$D + (E - this.$W) : E;
        if (F === y || F === L) {
          var X = this.clone().set(G, 1);
          X.$d[R](q), X.init(), this.$d = X.set(G, Math.min(this.$D, X.daysInMonth())).$d;
        } else
          R && this.$d[R](q);
        return this.init(), this;
      }, x.set = function(m, E) {
        return this.clone().$set(m, E);
      }, x.get = function(m) {
        return this[P.p(m)]();
      }, x.add = function(m, E) {
        var S, F = this;
        m = Number(m);
        var A = P.p(E), R = function(re) {
          var pe = _(F);
          return P.w(pe.date(pe.date() + Math.round(re * m)), F);
        };
        if (A === y)
          return this.set(y, this.$M + m);
        if (A === L)
          return this.set(L, this.$y + m);
        if (A === w)
          return R(1);
        if (A === j)
          return R(7);
        var q = (S = {}, S[c] = i, S[f] = n, S[V] = e, S)[A] || 1, X = this.$d.getTime() + m * q;
        return P.w(X, this);
      }, x.subtract = function(m, E) {
        return this.add(-1 * m, E);
      }, x.format = function(m) {
        var E = this, S = this.$locale();
        if (!this.isValid())
          return S.invalidDate || D;
        var F = m || "YYYY-MM-DDTHH:mm:ssZ", A = P.z(this), R = this.$H, q = this.$m, X = this.$M, re = S.weekdays, pe = S.months, me = function(oe, ie, ye, be) {
          return oe && (oe[ie] || oe(E, F)) || ye[ie].slice(0, be);
        }, xe = function(oe) {
          return P.s(R % 12 || 12, oe, "0");
        }, se = S.meridiem || function(oe, ie, ye) {
          var be = oe < 12 ? "AM" : "PM";
          return ye ? be.toLowerCase() : be;
        }, ve = { YY: String(this.$y).slice(-2), YYYY: this.$y, M: X + 1, MM: P.s(X + 1, 2, "0"), MMM: me(S.monthsShort, X, pe, 3), MMMM: me(pe, X), D: this.$D, DD: P.s(this.$D, 2, "0"), d: String(this.$W), dd: me(S.weekdaysMin, this.$W, re, 2), ddd: me(S.weekdaysShort, this.$W, re, 3), dddd: re[this.$W], H: String(R), HH: P.s(R, 2, "0"), h: xe(1), hh: xe(2), a: se(R, q, !0), A: se(R, q, !1), m: String(q), mm: P.s(q, 2, "0"), s: String(this.$s), ss: P.s(this.$s, 2, "0"), SSS: P.s(this.$ms, 3, "0"), Z: A };
        return F.replace(v, function(oe, ie) {
          return ie || ve[oe] || A.replace(":", "");
        });
      }, x.utcOffset = function() {
        return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
      }, x.diff = function(m, E, S) {
        var F, A = P.p(E), R = _(m), q = (R.utcOffset() - this.utcOffset()) * i, X = this - R, re = P.m(this, R);
        return re = (F = {}, F[L] = re / 12, F[y] = re, F[$] = re / 3, F[j] = (X - q) / 6048e5, F[w] = (X - q) / 864e5, F[f] = X / n, F[c] = X / i, F[V] = X / e, F)[A] || X, S ? re : P.a(re);
      }, x.daysInMonth = function() {
        return this.endOf(y).$D;
      }, x.$locale = function() {
        return r[this.$L];
      }, x.locale = function(m, E) {
        if (!m)
          return this.$L;
        var S = this.clone(), F = H(m, E, !0);
        return F && (S.$L = F), S;
      }, x.clone = function() {
        return P.w(this.$d, this);
      }, x.toDate = function() {
        return new Date(this.valueOf());
      }, x.toJSON = function() {
        return this.isValid() ? this.toISOString() : null;
      }, x.toISOString = function() {
        return this.$d.toISOString();
      }, x.toString = function() {
        return this.$d.toUTCString();
      }, O;
    }(), J = Z.prototype;
    return _.prototype = J, [["$ms", u], ["$s", V], ["$m", c], ["$H", f], ["$W", w], ["$M", y], ["$y", L], ["$D", G]].forEach(function(O) {
      J[O[1]] = function(x) {
        return this.$g(x, O[0], O[1]);
      };
    }), _.extend = function(O, x) {
      return O.$i || (O(x, Z, _), O.$i = !0), _;
    }, _.locale = H, _.isDayjs = M, _.unix = function(O) {
      return _(1e3 * O);
    }, _.en = r[l], _.Ls = r, _.p = {}, _;
  });
})(bt);
const s = bt.exports;
var xt = { exports: {} };
(function(t, a) {
  (function(e, i) {
    t.exports = i();
  })(Oe, function() {
    return function(e, i, n) {
      var u = i.prototype, V = function(y) {
        return y && (y.indexOf ? y : y.s);
      }, c = function(y, $, L, G, D) {
        var b = y.name ? y : y.$locale(), v = V(b[$]), g = V(b[L]), T = v || g.map(function(l) {
          return l.slice(0, G);
        });
        if (!D)
          return T;
        var B = b.weekStart;
        return T.map(function(l, r) {
          return T[(r + (B || 0)) % 7];
        });
      }, f = function() {
        return n.Ls[n.locale()];
      }, w = function(y, $) {
        return y.formats[$] || function(L) {
          return L.replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(G, D, b) {
            return D || b.slice(1);
          });
        }(y.formats[$.toUpperCase()]);
      }, j = function() {
        var y = this;
        return { months: function($) {
          return $ ? $.format("MMMM") : c(y, "months");
        }, monthsShort: function($) {
          return $ ? $.format("MMM") : c(y, "monthsShort", "months", 3);
        }, firstDayOfWeek: function() {
          return y.$locale().weekStart || 0;
        }, weekdays: function($) {
          return $ ? $.format("dddd") : c(y, "weekdays");
        }, weekdaysMin: function($) {
          return $ ? $.format("dd") : c(y, "weekdaysMin", "weekdays", 2);
        }, weekdaysShort: function($) {
          return $ ? $.format("ddd") : c(y, "weekdaysShort", "weekdays", 3);
        }, longDateFormat: function($) {
          return w(y.$locale(), $);
        }, meridiem: this.$locale().meridiem, ordinal: this.$locale().ordinal };
      };
      u.localeData = function() {
        return j.bind(this)();
      }, n.localeData = function() {
        var y = f();
        return { firstDayOfWeek: function() {
          return y.weekStart || 0;
        }, weekdays: function() {
          return n.weekdays();
        }, weekdaysShort: function() {
          return n.weekdaysShort();
        }, weekdaysMin: function() {
          return n.weekdaysMin();
        }, months: function() {
          return n.months();
        }, monthsShort: function() {
          return n.monthsShort();
        }, longDateFormat: function($) {
          return w(y, $);
        }, meridiem: y.meridiem, ordinal: y.ordinal };
      }, n.months = function() {
        return c(f(), "months");
      }, n.monthsShort = function() {
        return c(f(), "monthsShort", "months", 3);
      }, n.weekdays = function(y) {
        return c(f(), "weekdays", null, null, y);
      }, n.weekdaysShort = function(y) {
        return c(f(), "weekdaysShort", "weekdays", 3, y);
      }, n.weekdaysMin = function(y) {
        return c(f(), "weekdaysMin", "weekdays", 2, y);
      };
    };
  });
})(xt);
const Vr = xt.exports;
var gt = { exports: {} };
(function(t, a) {
  (function(e, i) {
    t.exports = i();
  })(Oe, function() {
    var e = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" };
    return function(i, n, u) {
      var V = n.prototype, c = V.format;
      u.en.formats = e, V.format = function(f) {
        f === void 0 && (f = "YYYY-MM-DDTHH:mm:ssZ");
        var w = this.$locale().formats, j = function(y, $) {
          return y.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(L, G, D) {
            var b = D && D.toUpperCase();
            return G || $[D] || e[D] || $[b].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(v, g, T) {
              return g || T.slice(1);
            });
          });
        }(f, w === void 0 ? {} : w);
        return c.call(this, j);
      };
    };
  });
})(gt);
const Pr = gt.exports;
var kt = { exports: {} };
(function(t, a) {
  (function(e, i) {
    t.exports = i();
  })(Oe, function() {
    var e = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" }, i = /(\[[^[]*\])|([-_:/.,()\s]+)|(A|a|YYYY|YY?|MM?M?M?|Do|DD?|hh?|HH?|mm?|ss?|S{1,3}|z|ZZ?)/g, n = /\d\d/, u = /\d\d?/, V = /\d*[^-_:/,()\s\d]+/, c = {}, f = function(D) {
      return (D = +D) + (D > 68 ? 1900 : 2e3);
    }, w = function(D) {
      return function(b) {
        this[D] = +b;
      };
    }, j = [/[+-]\d\d:?(\d\d)?|Z/, function(D) {
      (this.zone || (this.zone = {})).offset = function(b) {
        if (!b || b === "Z")
          return 0;
        var v = b.match(/([+-]|\d\d)/g), g = 60 * v[1] + (+v[2] || 0);
        return g === 0 ? 0 : v[0] === "+" ? -g : g;
      }(D);
    }], y = function(D) {
      var b = c[D];
      return b && (b.indexOf ? b : b.s.concat(b.f));
    }, $ = function(D, b) {
      var v, g = c.meridiem;
      if (g) {
        for (var T = 1; T <= 24; T += 1)
          if (D.indexOf(g(T, 0, b)) > -1) {
            v = T > 12;
            break;
          }
      } else
        v = D === (b ? "pm" : "PM");
      return v;
    }, L = { A: [V, function(D) {
      this.afternoon = $(D, !1);
    }], a: [V, function(D) {
      this.afternoon = $(D, !0);
    }], S: [/\d/, function(D) {
      this.milliseconds = 100 * +D;
    }], SS: [n, function(D) {
      this.milliseconds = 10 * +D;
    }], SSS: [/\d{3}/, function(D) {
      this.milliseconds = +D;
    }], s: [u, w("seconds")], ss: [u, w("seconds")], m: [u, w("minutes")], mm: [u, w("minutes")], H: [u, w("hours")], h: [u, w("hours")], HH: [u, w("hours")], hh: [u, w("hours")], D: [u, w("day")], DD: [n, w("day")], Do: [V, function(D) {
      var b = c.ordinal, v = D.match(/\d+/);
      if (this.day = v[0], b)
        for (var g = 1; g <= 31; g += 1)
          b(g).replace(/\[|\]/g, "") === D && (this.day = g);
    }], M: [u, w("month")], MM: [n, w("month")], MMM: [V, function(D) {
      var b = y("months"), v = (y("monthsShort") || b.map(function(g) {
        return g.slice(0, 3);
      })).indexOf(D) + 1;
      if (v < 1)
        throw new Error();
      this.month = v % 12 || v;
    }], MMMM: [V, function(D) {
      var b = y("months").indexOf(D) + 1;
      if (b < 1)
        throw new Error();
      this.month = b % 12 || b;
    }], Y: [/[+-]?\d+/, w("year")], YY: [n, function(D) {
      this.year = f(D);
    }], YYYY: [/\d{4}/, w("year")], Z: j, ZZ: j };
    function G(D) {
      var b, v;
      b = D, v = c && c.formats;
      for (var g = (D = b.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(_, P, Z) {
        var J = Z && Z.toUpperCase();
        return P || v[Z] || e[Z] || v[J].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(O, x, m) {
          return x || m.slice(1);
        });
      })).match(i), T = g.length, B = 0; B < T; B += 1) {
        var l = g[B], r = L[l], M = r && r[0], H = r && r[1];
        g[B] = H ? { regex: M, parser: H } : l.replace(/^\[|\]$/g, "");
      }
      return function(_) {
        for (var P = {}, Z = 0, J = 0; Z < T; Z += 1) {
          var O = g[Z];
          if (typeof O == "string")
            J += O.length;
          else {
            var x = O.regex, m = O.parser, E = _.slice(J), S = x.exec(E)[0];
            m.call(P, S), _ = _.replace(S, "");
          }
        }
        return function(F) {
          var A = F.afternoon;
          if (A !== void 0) {
            var R = F.hours;
            A ? R < 12 && (F.hours += 12) : R === 12 && (F.hours = 0), delete F.afternoon;
          }
        }(P), P;
      };
    }
    return function(D, b, v) {
      v.p.customParseFormat = !0, D && D.parseTwoDigitYear && (f = D.parseTwoDigitYear);
      var g = b.prototype, T = g.parse;
      g.parse = function(B) {
        var l = B.date, r = B.utc, M = B.args;
        this.$u = r;
        var H = M[1];
        if (typeof H == "string") {
          var _ = M[2] === !0, P = M[3] === !0, Z = _ || P, J = M[2];
          P && (J = M[2]), c = this.$locale(), !_ && J && (c = v.Ls[J]), this.$d = function(E, S, F) {
            try {
              if (["x", "X"].indexOf(S) > -1)
                return new Date((S === "X" ? 1e3 : 1) * E);
              var A = G(S)(E), R = A.year, q = A.month, X = A.day, re = A.hours, pe = A.minutes, me = A.seconds, xe = A.milliseconds, se = A.zone, ve = new Date(), oe = X || (R || q ? 1 : ve.getDate()), ie = R || ve.getFullYear(), ye = 0;
              R && !q || (ye = q > 0 ? q - 1 : ve.getMonth());
              var be = re || 0, Ae = pe || 0, Le = me || 0, Be = xe || 0;
              return se ? new Date(Date.UTC(ie, ye, oe, be, Ae, Le, Be + 60 * se.offset * 1e3)) : F ? new Date(Date.UTC(ie, ye, oe, be, Ae, Le, Be)) : new Date(ie, ye, oe, be, Ae, Le, Be);
            } catch {
              return new Date("");
            }
          }(l, H, r), this.init(), J && J !== !0 && (this.$L = this.locale(J).$L), Z && l != this.format(H) && (this.$d = new Date("")), c = {};
        } else if (H instanceof Array)
          for (var O = H.length, x = 1; x <= O; x += 1) {
            M[1] = H[x - 1];
            var m = v.apply(this, M);
            if (m.isValid()) {
              this.$d = m.$d, this.$L = m.$L, this.init();
              break;
            }
            x === O && (this.$d = new Date(""));
          }
        else
          T.call(this, B);
      };
    };
  });
})(kt);
const Cr = kt.exports;
var wt = { exports: {} };
(function(t, a) {
  (function(e, i) {
    t.exports = i();
  })(Oe, function() {
    return function(e, i, n) {
      i.prototype.isToday = function() {
        var u = "YYYY-MM-DD", V = n();
        return this.format(u) === V.format(u);
      };
    };
  });
})(wt);
const Yr = wt.exports;
var $t = { exports: {} };
(function(t, a) {
  (function(e, i) {
    t.exports = i();
  })(Oe, function() {
    return function(e, i, n) {
      i.prototype.isBetween = function(u, V, c, f) {
        var w = n(u), j = n(V), y = (f = f || "()")[0] === "(", $ = f[1] === ")";
        return (y ? this.isAfter(w, c) : !this.isBefore(w, c)) && ($ ? this.isBefore(j, c) : !this.isAfter(j, c)) || (y ? this.isBefore(w, c) : !this.isAfter(w, c)) && ($ ? this.isAfter(j, c) : !this.isBefore(j, c));
      };
    };
  });
})($t);
const Tr = $t.exports;
var jt = { exports: {} };
(function(t, a) {
  (function(e, i) {
    t.exports = i();
  })(Oe, function() {
    var e, i, n = 1e3, u = 6e4, V = 36e5, c = 864e5, f = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, w = 31536e6, j = 2592e6, y = /^(-|\+)?P(?:([-+]?[0-9,.]*)Y)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9,.]*)W)?(?:([-+]?[0-9,.]*)D)?(?:T(?:([-+]?[0-9,.]*)H)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9,.]*)S)?)?$/, $ = { years: w, months: j, days: c, hours: V, minutes: u, seconds: n, milliseconds: 1, weeks: 6048e5 }, L = function(l) {
      return l instanceof B;
    }, G = function(l, r, M) {
      return new B(l, M, r.$l);
    }, D = function(l) {
      return i.p(l) + "s";
    }, b = function(l) {
      return l < 0;
    }, v = function(l) {
      return b(l) ? Math.ceil(l) : Math.floor(l);
    }, g = function(l) {
      return Math.abs(l);
    }, T = function(l, r) {
      return l ? b(l) ? { negative: !0, format: "" + g(l) + r } : { negative: !1, format: "" + l + r } : { negative: !1, format: "" };
    }, B = function() {
      function l(M, H, _) {
        var P = this;
        if (this.$d = {}, this.$l = _, M === void 0 && (this.$ms = 0, this.parseFromMilliseconds()), H)
          return G(M * $[D(H)], this);
        if (typeof M == "number")
          return this.$ms = M, this.parseFromMilliseconds(), this;
        if (typeof M == "object")
          return Object.keys(M).forEach(function(O) {
            P.$d[D(O)] = M[O];
          }), this.calMilliseconds(), this;
        if (typeof M == "string") {
          var Z = M.match(y);
          if (Z) {
            var J = Z.slice(2).map(function(O) {
              return O != null ? Number(O) : 0;
            });
            return this.$d.years = J[0], this.$d.months = J[1], this.$d.weeks = J[2], this.$d.days = J[3], this.$d.hours = J[4], this.$d.minutes = J[5], this.$d.seconds = J[6], this.calMilliseconds(), this;
          }
        }
        return this;
      }
      var r = l.prototype;
      return r.calMilliseconds = function() {
        var M = this;
        this.$ms = Object.keys(this.$d).reduce(function(H, _) {
          return H + (M.$d[_] || 0) * $[_];
        }, 0);
      }, r.parseFromMilliseconds = function() {
        var M = this.$ms;
        this.$d.years = v(M / w), M %= w, this.$d.months = v(M / j), M %= j, this.$d.days = v(M / c), M %= c, this.$d.hours = v(M / V), M %= V, this.$d.minutes = v(M / u), M %= u, this.$d.seconds = v(M / n), M %= n, this.$d.milliseconds = M;
      }, r.toISOString = function() {
        var M = T(this.$d.years, "Y"), H = T(this.$d.months, "M"), _ = +this.$d.days || 0;
        this.$d.weeks && (_ += 7 * this.$d.weeks);
        var P = T(_, "D"), Z = T(this.$d.hours, "H"), J = T(this.$d.minutes, "M"), O = this.$d.seconds || 0;
        this.$d.milliseconds && (O += this.$d.milliseconds / 1e3);
        var x = T(O, "S"), m = M.negative || H.negative || P.negative || Z.negative || J.negative || x.negative, E = Z.format || J.format || x.format ? "T" : "", S = (m ? "-" : "") + "P" + M.format + H.format + P.format + E + Z.format + J.format + x.format;
        return S === "P" || S === "-P" ? "P0D" : S;
      }, r.toJSON = function() {
        return this.toISOString();
      }, r.format = function(M) {
        var H = M || "YYYY-MM-DDTHH:mm:ss", _ = { Y: this.$d.years, YY: i.s(this.$d.years, 2, "0"), YYYY: i.s(this.$d.years, 4, "0"), M: this.$d.months, MM: i.s(this.$d.months, 2, "0"), D: this.$d.days, DD: i.s(this.$d.days, 2, "0"), H: this.$d.hours, HH: i.s(this.$d.hours, 2, "0"), m: this.$d.minutes, mm: i.s(this.$d.minutes, 2, "0"), s: this.$d.seconds, ss: i.s(this.$d.seconds, 2, "0"), SSS: i.s(this.$d.milliseconds, 3, "0") };
        return H.replace(f, function(P, Z) {
          return Z || String(_[P]);
        });
      }, r.as = function(M) {
        return this.$ms / $[D(M)];
      }, r.get = function(M) {
        var H = this.$ms, _ = D(M);
        return _ === "milliseconds" ? H %= 1e3 : H = _ === "weeks" ? v(H / $[_]) : this.$d[_], H === 0 ? 0 : H;
      }, r.add = function(M, H, _) {
        var P;
        return P = H ? M * $[D(H)] : L(M) ? M.$ms : G(M, this).$ms, G(this.$ms + P * (_ ? -1 : 1), this);
      }, r.subtract = function(M, H) {
        return this.add(M, H, !0);
      }, r.locale = function(M) {
        var H = this.clone();
        return H.$l = M, H;
      }, r.clone = function() {
        return G(this.$ms, this);
      }, r.humanize = function(M) {
        return e().add(this.$ms, "ms").locale(this.$l).fromNow(!M);
      }, r.milliseconds = function() {
        return this.get("milliseconds");
      }, r.asMilliseconds = function() {
        return this.as("milliseconds");
      }, r.seconds = function() {
        return this.get("seconds");
      }, r.asSeconds = function() {
        return this.as("seconds");
      }, r.minutes = function() {
        return this.get("minutes");
      }, r.asMinutes = function() {
        return this.as("minutes");
      }, r.hours = function() {
        return this.get("hours");
      }, r.asHours = function() {
        return this.as("hours");
      }, r.days = function() {
        return this.get("days");
      }, r.asDays = function() {
        return this.as("days");
      }, r.weeks = function() {
        return this.get("weeks");
      }, r.asWeeks = function() {
        return this.as("weeks");
      }, r.months = function() {
        return this.get("months");
      }, r.asMonths = function() {
        return this.as("months");
      }, r.years = function() {
        return this.get("years");
      }, r.asYears = function() {
        return this.as("years");
      }, l;
    }();
    return function(l, r, M) {
      e = M, i = M().$utils(), M.duration = function(P, Z) {
        var J = M.locale();
        return G(P, { $l: J }, Z);
      }, M.isDuration = L;
      var H = r.prototype.add, _ = r.prototype.subtract;
      r.prototype.add = function(P, Z) {
        return L(P) && (P = P.asMilliseconds()), H.bind(this)(P, Z);
      }, r.prototype.subtract = function(P, Z) {
        return L(P) && (P = P.asMilliseconds()), _.bind(this)(P, Z);
      };
    };
  });
})(jt);
const Ar = jt.exports;
function Lr() {
  const t = (c) => {
    const f = [], w = c.localeData().firstDayOfWeek();
    for (let j = 0; j <= c.date(0 - w).day(); j++)
      f.push(c.date(0).subtract(j, "day"));
    return f.sort((j, y) => j.date() - y.date());
  };
  return {
    usePreviousDate: t,
    useCurrentDate: (c) => Array.from(
      {
        length: c.daysInMonth()
      },
      (f, w) => c.date(w + 1)
    ),
    useNextDate: (c) => {
      const f = [];
      for (let w = 1; w <= 42 - (t(c).length + c.daysInMonth()); w++)
        f.push(c.date(w).month(c.month()).add(1, "month"));
      return f;
    },
    useDisableDate: (c, { disableDate: f }) => typeof f == "function" ? f(c.toDate()) : !1,
    useBetweenRange: (c, { previous: f, next: w }) => {
      let j;
      return f.isAfter(w, "date") ? j = "(]" : j = "[)", !!(c.isBetween(f, w, "date", j) && !c.off);
    },
    useToValueFromString: (c, { formatter: f }) => c.format(f.date),
    useToValueFromArray: ({ previous: c, next: f }, { formatter: w, separator: j }) => `${c.format(w.date)}${j}${f.format(w.date)}`
  };
}
function Br() {
  return {
    useVisibleViewport: (a) => {
      if (a) {
        const { right: e } = a.getBoundingClientRect(), i = window.innerWidth || document.documentElement.clientWidth;
        return e > i;
      } else
        return null;
    }
  };
}
const Er = ["placeholder"], Fr = { class: "absolute inset-y-0 right-0 inline-flex items-center rounded-md overflow-hidden" }, Nr = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, Hr = {
  key: 0,
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "1.5",
  d: "M6 18L18 6M6 6l12 12"
}, Rr = {
  key: 1,
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "1.5",
  d: "M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
}, Ir = { class: "flex flex-wrap lg:flex-nowrap" }, Ur = { class: "relative flex flex-wrap sm:flex-nowrap p-1" }, zr = {
  key: 0,
  class: "hidden h-full absolute inset-0 sm:flex justify-center items-center"
}, Wr = /* @__PURE__ */ C("div", { class: "h-full border-r border-black/[.1] dark:border-vtd-secondary-700/[1]" }, null, -1), Zr = [
  Wr
], Gr = { class: "px-0.5 sm:px-2" }, Kr = {
  key: 1,
  class: "relative w-full sm:w-80 overflow-hidden mt-3 sm:mt-0 sm:ml-2"
}, Jr = { class: "px-0.5 sm:px-2" }, qr = { key: 0 }, Xr = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, Qr = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, eo = ["disabled", "onClick", "textContent"], to = ["onClick", "textContent"], ro = {
  key: 1,
  class: "sm:hidden"
}, oo = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, ao = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, no = ["onClick", "textContent"], so = {
  key: 1,
  class: "flex"
}, lo = { class: "bg-white rounded-lg shadow-sm border-0 border border-black/[.1] px-3 py-3 sm:px-4 sm:py-4 dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700/[1]" }, io = { class: "flex flex-wrap lg:flex-nowrap" }, uo = { class: "relative flex flex-wrap sm:flex-nowrap p-1" }, co = {
  key: 0,
  class: "hidden h-full absolute inset-0 sm:flex justify-center items-center"
}, fo = /* @__PURE__ */ C("div", { class: "h-full border-r border-black/[.1] dark:border-vtd-secondary-700/[1]" }, null, -1), mo = [
  fo
], po = { class: "px-0.5 sm:px-2" }, vo = {
  key: 1,
  class: "relative w-full sm:w-80 overflow-hidden mt-3 sm:mt-0 sm:ml-2"
}, ho = { class: "px-0.5 sm:px-2" }, yo = { key: 0 }, bo = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, xo = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, go = ["disabled", "textContent"], ko = {
  __name: "VueTailwindDatePicker",
  props: {
    noInput: Boolean,
    overlay: Boolean,
    asSingle: Boolean,
    useRange: Boolean,
    placeholder: {
      type: [Boolean, String],
      default: !1
    },
    i18n: {
      type: String,
      default: "en"
    },
    inputClasses: {
      type: String,
      default: ""
    },
    disableInRange: {
      type: Boolean,
      default: !0
    },
    disableDate: {
      type: [Boolean, Array, Function],
      default: !1
    },
    autoApply: {
      type: Boolean,
      default: !0
    },
    shortcuts: {
      type: [Boolean, Function],
      default: !0
    },
    separator: {
      type: String,
      default: " ~ "
    },
    formatter: {
      type: Object,
      default: () => ({
        date: "YYYY-MM-DD HH:mm:ss",
        month: "MMM"
      })
    },
    modelValue: {
      type: [Array, Object, String],
      default: () => []
    },
    startFrom: {
      type: [Object, String],
      default: () => new Date()
    },
    options: {
      type: Object,
      default: () => ({
        shortcuts: {
          today: "Today",
          yesterday: "Yesterday",
          past: (t) => `Last ${t} Days`,
          currentMonth: "This Month",
          pastMonth: "Last Month"
        },
        footer: {
          apply: "Apply",
          cancel: "Cancel"
        }
      })
    }
  },
  emits: ["update:modelValue"],
  setup(t, { emit: a }) {
    const e = t, {
      useCurrentDate: i,
      useDisableDate: n,
      useBetweenRange: u,
      useNextDate: V,
      usePreviousDate: c,
      useToValueFromArray: f,
      useToValueFromString: w
    } = Lr(), { useVisibleViewport: j } = Br();
    s.extend(Vr), s.extend(Pr), s.extend(Cr), s.extend(Yr), s.extend(Tr), s.extend(Ar);
    const y = te(null), $ = te(null), L = te(null), G = te(""), D = te(null), b = te(""), v = te([]), g = te([]), T = te(null), B = te(null), l = _t({
      previous: {
        calendar: !0,
        month: !1,
        year: !1
      },
      next: {
        calendar: !0,
        month: !1,
        year: !1
      }
    }), r = te({
      previous: s(),
      next: s().add(1, "month"),
      year: {
        previous: s().year(),
        next: s().year()
      },
      weeks: s.weekdaysShort(),
      months: e.formatter.month === "MMM" ? s.monthsShort() : s.months()
    }), M = fe(() => r.value.weeks), H = fe(() => r.value.months), _ = fe(() => {
      const { previous: d, next: h, year: k } = N(r);
      return {
        previous: {
          date: () => c(d).concat(i(d)).concat(V(d)).map((o) => (o.today = o.isToday(), o.active = d.month() === o.month(), o.off = d.month() !== o.month(), o.sunday = o.day() === 0, o.disabled = n(o, e) && !E(o), o.inRange = () => {
            if (e.asSingle && !e.useRange)
              return d.month() !== o.month();
          }, o.hovered = () => m() && v.value.length > 1 ? (o.isBetween(v.value[0], v.value[1], "date", "()") || o.isBetween(v.value[1], v.value[0], "date", "(]")) && d.month() === o.month() : !1, o.duration = () => !1, o)),
          month: d && d.format(e.formatter.month),
          year: d && d.year(),
          years: () => Array.from(
            {
              length: 12
            },
            (o, p) => k.previous + p
          ),
          onPrevious: () => {
            r.value.previous = d.subtract(1, "month");
          },
          onNext: () => {
            r.value.previous = d.add(1, "month"), d.diff(h, "month") === -1 && (r.value.next = h.add(1, "month"));
          },
          onPreviousYear: () => {
            r.value.year.previous = r.value.year.previous - 12;
          },
          onNextYear: () => {
            r.value.year.previous = r.value.year.previous + 12;
          },
          openMonth: () => {
            l.previous.month = !l.previous.month, l.previous.year = !1, l.previous.calendar = !l.previous.month;
          },
          setMount: (o) => {
            r.value.previous = d.month(o), l.previous.month = !l.previous.month, l.previous.year = !1, l.previous.calendar = !l.previous.month, Ee(() => {
              (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month")), r.value.year.next = r.value.next.year();
            });
          },
          openYear: () => {
            l.previous.year = !l.previous.year, l.previous.month = !1, l.previous.calendar = !l.previous.year;
          },
          setYear: (o, p) => {
            p || (r.value.previous = d.year(o), l.previous.year = !l.previous.year, l.previous.calendar = !l.previous.year, Ee(() => {
              (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month")), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year();
            }));
          }
        },
        next: {
          date: () => c(h).concat(i(h)).concat(V(h)).map((o) => (o.today = o.isToday(), o.active = h.month() === o.month(), o.off = h.month() !== o.month(), o.sunday = o.day() === 0, o.disabled = n(o, e) && !E(o), o.inRange = () => {
            if (e.asSingle && !e.useRange)
              return h.month() !== o.month();
          }, o.hovered = () => v.value.length > 1 ? (o.isBetween(v.value[0], v.value[1], "date", "()") || o.isBetween(v.value[1], v.value[0], "date", "(]")) && h.month() === o.month() : !1, o.duration = () => !1, o)),
          month: h && h.format(e.formatter.month),
          year: h && h.year(),
          years: () => Array.from(
            {
              length: 12
            },
            (o, p) => k.next + p
          ),
          onPrevious: () => {
            r.value.next = h.subtract(1, "month"), h.diff(d, "month") === 1 && (r.value.previous = d.subtract(1, "month"));
          },
          onNext: () => {
            r.value.next = h.add(1, "month");
          },
          onPreviousYear: () => {
            r.value.year.next = r.value.year.next - 12;
          },
          onNextYear: () => {
            r.value.year.next = r.value.year.next + 12;
          },
          openMonth: () => {
            l.next.month = !l.next.month, l.next.year = !1, l.next.calendar = !l.next.month;
          },
          setMount: (o) => {
            r.value.next = h.month(o), l.next.month = !l.next.month, l.next.year = !1, l.next.calendar = !l.next.month, Ee(() => {
              (r.value.previous.isSame(r.value.next, "month") || r.value.previous.isAfter(r.value.next)) && (r.value.previous = r.value.next.subtract(1, "month")), r.value.year.previous = r.value.previous.year();
            });
          },
          openYear: () => {
            l.next.year = !l.next.year, l.next.month = !1, l.next.calendar = !l.next.year;
          },
          setYear: (o, p) => {
            p && (r.value.next = h.year(o), l.next.year = !l.next.year, l.next.month = !1, l.next.calendar = !l.next.year, Ee(() => {
              (r.value.previous.isSame(r.value.next, "month") || r.value.previous.isAfter(r.value.next)) && (r.value.previous = r.value.next.subtract(1, "month")), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year();
            }));
          }
        }
      };
    }), P = te(!1);
    setTimeout(() => {
      P.value = !0;
    }, 250);
    const Z = () => s().localeData().firstDayOfWeek(), J = (d) => {
      const h = [...d], k = h.shift();
      return [...h, k];
    }, O = () => Array.isArray(e.modelValue), x = () => typeof e.modelValue == "object", m = () => !e.useRange && !e.asSingle ? !0 : !e.useRange && e.asSingle ? !1 : e.useRange && !e.asSingle ? !0 : !!(e.useRange && e.asSingle), E = (d) => {
      if (e.disableInRange || b.value === "")
        return !1;
      let h, k;
      if (O()) {
        const [o, p] = e.modelValue;
        h = o, k = p;
      } else if (x()) {
        if (e.modelValue) {
          const [o, p] = Object.values(e.modelValue);
          h = o, k = p;
        }
      } else {
        const [o, p] = e.modelValue.split(e.separator);
        h = o, k = p;
      }
      return d.isBetween(s(h, e.formatter.date, !0), s(k, e.formatter.date, !0), "date", "[]");
    }, S = () => {
      T.value = null, B.value = null, v.value = [], D.value = null;
    }, F = () => {
      if (b.value = "", O())
        a("update:modelValue", []);
      else if (x()) {
        const d = {}, [h, k] = Object.keys(e.modelValue);
        d[h] = "", d[k] = "", a("update:modelValue", d);
      } else
        a("update:modelValue", "");
      g.value = [], $.value && $.value.focus();
    }, A = () => {
      if (m()) {
        const [d, h] = b.value.split(e.separator), [k, o] = [s(d, e.formatter.date, !0), s(h, e.formatter.date, !0)];
        if (k.isValid() && o.isValid())
          if (R(k), R(o), O())
            a("update:modelValue", [d, h]);
          else if (x()) {
            const p = {}, [Y, z] = Object.keys(e.modelValue);
            p[Y] = d, p[z] = h, a("update:modelValue", p);
          } else
            a(
              "update:modelValue",
              f(
                {
                  previous: k,
                  next: o
                },
                e
              )
            );
      } else {
        const d = s(b.value, e.formatter.date, !0);
        if (d.isValid())
          if (R(d), O())
            a("update:modelValue", [b.value]);
          else if (x()) {
            const h = {}, [k] = Object.keys(e.modelValue);
            h[k] = b.value, a("update:modelValue", h);
          } else
            a("update:modelValue", b.value);
      }
    }, R = (d, h, k) => {
      if (m())
        if (T.value)
          if (B.value = d, e.autoApply) {
            d.isBefore(T.value) ? b.value = f(
              {
                previous: d,
                next: T.value
              },
              e
            ) : b.value = f(
              {
                previous: T.value,
                next: d
              },
              e
            );
            const [o, p] = b.value.split(e.separator);
            if (O())
              a("update:modelValue", [
                s(o, e.formatter.date, !0).format(e.formatter.date),
                s(p, e.formatter.date, !0).format(e.formatter.date)
              ]);
            else if (x()) {
              const Y = {}, [z, ae] = Object.keys(e.modelValue);
              Y[z] = o, Y[ae] = p, a("update:modelValue", Y);
            } else
              a(
                "update:modelValue",
                f(
                  {
                    previous: s(o, e.formatter.date, !0),
                    next: s(p, e.formatter.date, !0)
                  },
                  e
                )
              );
            k && k(), g.value = [], s(o, e.formatter.date, !0).isSame(s(p, e.formatter.date, !0), "month") || (r.value.previous = s(o, e.formatter.date, !0), r.value.next = s(p, e.formatter.date, !0)), S();
          } else {
            T.value.isAfter(d, "month") ? g.value = [d, T.value] : g.value = [T.value, d];
            const [o, p] = g.value;
            o.isSame(p, "month") || (r.value.previous = o, r.value.next = p), S();
          }
        else
          g.value = [], T.value = d, D.value = d, v.value.push(d), g.value.push(d), h ? (r.value.next = d, r.value.previous.isSame(d, "month") && (r.value.next = d.add(1, "month"))) : (r.value.previous = d, r.value.next.isSame(d, "month") && (r.value.previous = r.value.next, r.value.next = d.add(1, "month")));
      else if (e.autoApply) {
        if (b.value = w(d, e), O())
          a("update:modelValue", [b.value]);
        else if (x()) {
          const o = {}, [p] = Object.keys(e.modelValue);
          o[p] = b.value, a("update:modelValue", o);
        } else
          a("update:modelValue", b.value);
        k(), g.value = [], S();
      } else
        g.value = [d], S();
    }, q = (d) => {
      if (g.value.length < 1)
        return !1;
      let h;
      if (m()) {
        const [k, o] = g.value;
        o.isBefore(k) ? h = f(
          {
            previous: o,
            next: k
          },
          e
        ) : h = f(
          {
            previous: k,
            next: o
          },
          e
        );
      } else {
        const [k] = g.value;
        h = k;
      }
      if (m()) {
        const [k, o] = h.split(e.separator);
        if (O())
          a("update:modelValue", [
            s(k, e.formatter.date, !0).format(e.formatter.date),
            s(o, e.formatter.date, !0).format(e.formatter.date)
          ]);
        else if (x()) {
          const p = {}, [Y, z] = Object.keys(e.modelValue);
          p[Y] = k, p[z] = o, a("update:modelValue", p);
        } else
          a(
            "update:modelValue",
            f(
              {
                previous: s(k, e.formatter.date, !0),
                next: s(o, e.formatter.date, !0)
              },
              e
            )
          );
        b.value = h;
      } else if (b.value = h.format(e.formatter.date), O())
        a("update:modelValue", [b.value]);
      else if (x()) {
        const k = {}, [o] = Object.keys(e.modelValue);
        k[o] = b.value, a("update:modelValue", k);
      } else
        a("update:modelValue", b.value);
      d && d();
    }, X = (d) => {
      if (!m())
        return !1;
      if (T.value)
        v.value = [T.value, d];
      else
        return v.value = [], !1;
    }, re = (d) => {
      if (T.value && e.autoApply)
        return !1;
      let h, k;
      if (v.value.length > 1) {
        const [o, p] = v.value;
        h = s(o, e.formatter.date, !0), k = s(p, e.formatter.date, !0);
      } else if (O())
        if (e.autoApply) {
          const [o, p] = e.modelValue;
          h = o && s(o, e.formatter.date, !0), k = p && s(p, e.formatter.date, !0);
        } else {
          const [o, p] = g.value;
          h = s(o, e.formatter.date, !0), k = s(p, e.formatter.date, !0);
        }
      else if (x())
        if (e.autoApply) {
          if (e.modelValue) {
            const [o, p] = Object.values(e.modelValue);
            h = o && s(o, e.formatter.date, !0), k = p && s(p, e.formatter.date, !0);
          }
        } else {
          const [o, p] = g.value;
          h = s(o, e.formatter.date, !0), k = s(p, e.formatter.date, !0);
        }
      else if (e.autoApply) {
        const [o, p] = e.modelValue ? e.modelValue.split(e.separator) : [!1, !1];
        h = o && s(o, e.formatter.date, !0), k = p && s(p, e.formatter.date, !0);
      } else {
        const [o, p] = g.value;
        h = s(o, e.formatter.date, !0), k = s(p, e.formatter.date, !0);
      }
      return h && k ? u(d, {
        previous: h,
        next: k
      }) : !1;
    }, pe = (d) => {
      const { today: h, active: k, off: o, disabled: p } = d;
      let Y, z, ae;
      if (m())
        if (O())
          if (D.value) {
            const [I, Q] = v.value;
            z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
          } else if (e.autoApply) {
            const [I, Q] = e.modelValue;
            z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
          } else {
            const [I, Q] = g.value;
            z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
          }
        else if (x())
          if (D.value) {
            const [I, Q] = v.value;
            z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
          } else if (e.autoApply) {
            const [I, Q] = e.modelValue ? Object.values(e.modelValue) : [!1, !1];
            z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
          } else {
            const [I, Q] = g.value;
            z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
          }
        else if (D.value) {
          const [I, Q] = v.value;
          z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
        } else if (e.autoApply) {
          const [I, Q] = e.modelValue ? e.modelValue.split(e.separator) : [!1, !1];
          z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
        } else {
          const [I, Q] = g.value;
          z = I && s(I, e.formatter.date, !0), ae = Q && s(Q, e.formatter.date, !0);
        }
      else if (O())
        if (e.autoApply) {
          if (e.modelValue.length > 0) {
            const [I] = e.modelValue;
            z = s(I, e.formatter.date, !0);
          }
        } else {
          const [I] = g.value;
          z = I && s(I, e.formatter.date, !0);
        }
      else if (x())
        if (e.autoApply) {
          if (e.modelValue) {
            const [I] = Object.values(e.modelValue);
            z = s(I, e.formatter.date, !0);
          }
        } else {
          const [I] = g.value;
          z = I && s(I, e.formatter.date, !0);
        }
      else if (e.autoApply) {
        if (e.modelValue) {
          const [I] = e.modelValue.split(e.separator);
          z = s(I, e.formatter.date, !0);
        }
      } else {
        const [I] = g.value;
        z = I && s(I, e.formatter.date, !0);
      }
      return k && (Y = h ? "text-vtd-primary-500 font-semibold dark:text-vtd-primary-400 rounded-full" : p ? "text-vtd-secondary-600 font-normal disabled:text-vtd-secondary-500 disabled:cursor-not-allowed rounded-full" : d.isBetween(z, ae, "date", "()") ? "text-vtd-secondary-700 font-medium dark:text-vtd-secondary-100 rounded-full" : "text-vtd-secondary-600 font-medium dark:text-vtd-secondary-200 rounded-full"), o && (Y = "text-vtd-secondary-400 font-light disabled:cursor-not-allowed"), z && ae && !o ? (d.isSame(z, "date") && (Y = ae.isAfter(z, "date") ? "bg-vtd-primary-500 text-white font-bold rounded-l-full disabled:cursor-not-allowed" : "bg-vtd-primary-500 text-white font-bold rounded-r-full disabled:cursor-not-allowed", z.isSame(ae, "date") && (Y = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed")), d.isSame(ae, "date") && (Y = ae.isAfter(z, "date") ? "bg-vtd-primary-500 text-white font-bold rounded-r-full disabled:cursor-not-allowed" : "bg-vtd-primary-500 text-white font-bold rounded-l-full disabled:cursor-not-allowed", z.isSame(ae, "date") && (Y = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed"))) : z && d.isSame(z, "date") && !o && (Y = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed"), Y;
    }, me = (d) => {
      let h, k, o;
      if (h = "", !m())
        return h;
      if (O())
        if (v.value.length > 1) {
          const [p, Y] = v.value;
          k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
        } else if (e.autoApply) {
          const [p, Y] = e.modelValue;
          k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
        } else {
          const [p, Y] = g.value;
          k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
        }
      else if (x())
        if (v.value.length > 1) {
          const [p, Y] = v.value;
          k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
        } else if (e.autoApply) {
          if (e.modelValue) {
            const [p, Y] = Object.values(e.modelValue);
            k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
          }
        } else {
          const [p, Y] = g.value;
          k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
        }
      else if (v.value.length > 1) {
        const [p, Y] = v.value;
        k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
      } else if (e.autoApply) {
        const [p, Y] = e.modelValue ? e.modelValue.split(e.separator) : [!1, !1];
        k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
      } else {
        const [p, Y] = g.value;
        k = p && s(p, e.formatter.date, !0), o = Y && s(Y, e.formatter.date, !0);
      }
      return k && o && (d.isSame(k, "date") ? (o.isBefore(k) && (h += " rounded-r-full inset-0"), k.isBefore(o) && (h += " rounded-l-full inset-0")) : d.isSame(o, "date") ? (o.isBefore(k) && (h += " rounded-l-full inset-0"), k.isBefore(o) && (h += " rounded-r-full inset-0")) : h += " inset-0"), h;
    }, xe = (d, h) => {
      r.value.previous = s(d, e.formatter.date, !0), r.value.next = s(h, e.formatter.date, !0), (s.duration(r.value.next.diff(r.value.previous)).$d.months === 2 || s.duration(r.value.next.diff(r.value.previous)).$d.months === 1 && s.duration(r.value.next.diff(r.value.previous)).$d.days === 7) && (r.value.next = r.value.next.subtract(1, "month")), (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month"));
    }, se = (d, h) => {
      if (m())
        if (e.autoApply) {
          if (O())
            a("update:modelValue", [d, h]);
          else if (x()) {
            const k = {}, [o, p] = Object.keys(e.modelValue);
            k[o] = d, k[p] = h, a("update:modelValue", k);
          } else
            a(
              "update:modelValue",
              f(
                {
                  previous: s(d, e.formatter.date, !0),
                  next: s(h, e.formatter.date, !0)
                },
                e
              )
            );
          b.value = `${d}${e.separator}${h}`;
        } else
          g.value = [s(d, e.formatter.date, !0), s(h, e.formatter.date, !0)];
      else if (e.autoApply) {
        if (O())
          a("update:modelValue", [d]);
        else if (x()) {
          const k = {}, [o] = Object.keys(e.modelValue);
          k[o] = d, a("update:modelValue", k);
        } else
          a("update:modelValue", d);
        b.value = d, close();
      } else
        g.value = [s(d, e.formatter.date, !0), s(h, e.formatter.date, !0)];
      xe(d, h);
    }, ve = (d) => {
      const h = s().format(e.formatter.date), k = s().format(e.formatter.date);
      se(h, k), d();
    }, oe = (d) => {
      const h = s().subtract(1, "day").format(e.formatter.date), k = s().subtract(1, "day").format(e.formatter.date);
      se(h, k), d();
    }, ie = (d, h) => {
      const k = s().subtract(d - 1, "day").format(e.formatter.date), o = s().format(e.formatter.date);
      se(k, o), h();
    }, ye = (d) => {
      const h = s().date(1).format(e.formatter.date), k = s().date(s().daysInMonth()).format(e.formatter.date);
      se(h, k), d();
    }, be = (d) => {
      const h = s().date(1).subtract(1, "month").format(e.formatter.date), k = s().date(0).format(e.formatter.date);
      se(h, k), d();
    }, Ae = (d, h) => {
      let k, o;
      const [p, Y] = d.atClick();
      k = s(p).format(e.formatter.date), o = s(Y).format(e.formatter.date), se(k, o), h();
    };
    Vt(
      () => g.value,
      (d) => {
        d.length > 0 && (l.previous.calendar = !0, l.previous.month = !1, l.previous.year = !1, l.next.calendar = !0, l.next.month = !1, l.next.year = !1);
      }
    ), ge(() => {
      e.placeholder ? G.value = e.placeholder : m() ? G.value = `${e.formatter.date}${e.separator}${e.formatter.date}` : G.value = e.formatter.date;
    }), ge(() => {
      const d = e.i18n;
      Ee(() => {
        const h = /* @__PURE__ */ Object.assign({ "./locale/af.js": () => import("./af.a8e6019a.mjs"), "./locale/am.js": () => import("./am.c23dd6e7.mjs"), "./locale/ar-dz.js": () => import("./ar-dz.9496b9f2.mjs"), "./locale/ar-iq.js": () => import("./ar-iq.b00ab36a.mjs"), "./locale/ar-kw.js": () => import("./ar-kw.ed3c9e2e.mjs"), "./locale/ar-ly.js": () => import("./ar-ly.5f58fa8a.mjs"), "./locale/ar-ma.js": () => import("./ar-ma.62bec4e5.mjs"), "./locale/ar-sa.js": () => import("./ar-sa.8eab5439.mjs"), "./locale/ar-tn.js": () => import("./ar-tn.062a4c58.mjs"), "./locale/ar.js": () => import("./ar.3d9a1173.mjs"), "./locale/az.js": () => import("./az.82993bad.mjs"), "./locale/be.js": () => import("./be.8d8373e0.mjs"), "./locale/bg.js": () => import("./bg.9782490e.mjs"), "./locale/bi.js": () => import("./bi.defb6d51.mjs"), "./locale/bm.js": () => import("./bm.43bd9c42.mjs"), "./locale/bn-bd.js": () => import("./bn-bd.06bcf1b9.mjs"), "./locale/bn.js": () => import("./bn.70cd5cd6.mjs"), "./locale/bo.js": () => import("./bo.14da9654.mjs"), "./locale/br.js": () => import("./br.5f99088e.mjs"), "./locale/bs.js": () => import("./bs.10ccf088.mjs"), "./locale/ca.js": () => import("./ca.154c5633.mjs"), "./locale/cs.js": () => import("./cs.8c3dfde3.mjs"), "./locale/cv.js": () => import("./cv.19eb0897.mjs"), "./locale/cy.js": () => import("./cy.6648ced3.mjs"), "./locale/da.js": () => import("./da.e2e5b23d.mjs"), "./locale/de-at.js": () => import("./de-at.766825de.mjs"), "./locale/de-ch.js": () => import("./de-ch.be2f67a7.mjs"), "./locale/de.js": () => import("./de.9c5bd74e.mjs"), "./locale/dv.js": () => import("./dv.83ab9569.mjs"), "./locale/el.js": () => import("./el.53f7e115.mjs"), "./locale/en-au.js": () => import("./en-au.8da08692.mjs"), "./locale/en-ca.js": () => import("./en-ca.9328b555.mjs"), "./locale/en-gb.js": () => import("./en-gb.4add9a1e.mjs"), "./locale/en-ie.js": () => import("./en-ie.0dd48145.mjs"), "./locale/en-il.js": () => import("./en-il.9a00126c.mjs"), "./locale/en-in.js": () => import("./en-in.b30964ff.mjs"), "./locale/en-nz.js": () => import("./en-nz.c6222371.mjs"), "./locale/en-sg.js": () => import("./en-sg.30b1a07d.mjs"), "./locale/en-tt.js": () => import("./en-tt.41cda484.mjs"), "./locale/en.js": () => import("./en.c289298e.mjs"), "./locale/eo.js": () => import("./eo.c07f38f7.mjs"), "./locale/es-do.js": () => import("./es-do.ffe2f569.mjs"), "./locale/es-mx.js": () => import("./es-mx.653a6c79.mjs"), "./locale/es-pr.js": () => import("./es-pr.4e494088.mjs"), "./locale/es-us.js": () => import("./es-us.d8e9ee63.mjs"), "./locale/es.js": () => import("./es.94dc5c5d.mjs"), "./locale/et.js": () => import("./et.bb501f20.mjs"), "./locale/eu.js": () => import("./eu.3b168baa.mjs"), "./locale/fa.js": () => import("./fa.abedd215.mjs"), "./locale/fi.js": () => import("./fi.1f6db8d2.mjs"), "./locale/fo.js": () => import("./fo.a8c7a9b1.mjs"), "./locale/fr-ca.js": () => import("./fr-ca.c1bfb9f6.mjs"), "./locale/fr-ch.js": () => import("./fr-ch.c52ab391.mjs"), "./locale/fr.js": () => import("./fr.c04ca19c.mjs"), "./locale/fy.js": () => import("./fy.77090dfa.mjs"), "./locale/ga.js": () => import("./ga.5552a4b0.mjs"), "./locale/gd.js": () => import("./gd.6f1b8121.mjs"), "./locale/gl.js": () => import("./gl.1fa4164a.mjs"), "./locale/gom-latn.js": () => import("./gom-latn.cae13414.mjs"), "./locale/gu.js": () => import("./gu.20ed2ffd.mjs"), "./locale/he.js": () => import("./he.1bbb0faf.mjs"), "./locale/hi.js": () => import("./hi.1b255b5f.mjs"), "./locale/hr.js": () => import("./hr.369f1dee.mjs"), "./locale/ht.js": () => import("./ht.5d305bf9.mjs"), "./locale/hu.js": () => import("./hu.9b947a4f.mjs"), "./locale/hy-am.js": () => import("./hy-am.8043b7f1.mjs"), "./locale/id.js": () => import("./id.c8381924.mjs"), "./locale/is.js": () => import("./is.feb383e3.mjs"), "./locale/it-ch.js": () => import("./it-ch.05695138.mjs"), "./locale/it.js": () => import("./it.a293f8be.mjs"), "./locale/ja.js": () => import("./ja.19630e79.mjs"), "./locale/jv.js": () => import("./jv.6cf5da28.mjs"), "./locale/ka.js": () => import("./ka.0f337f81.mjs"), "./locale/kk.js": () => import("./kk.a19a821b.mjs"), "./locale/km.js": () => import("./km.f0227b9d.mjs"), "./locale/kn.js": () => import("./kn.b50902aa.mjs"), "./locale/ko.js": () => import("./ko.3fb54fd6.mjs"), "./locale/ku.js": () => import("./ku.be59a022.mjs"), "./locale/ky.js": () => import("./ky.c3bcfa2d.mjs"), "./locale/lb.js": () => import("./lb.2c48f206.mjs"), "./locale/lo.js": () => import("./lo.d73e6443.mjs"), "./locale/lt.js": () => import("./lt.29dd7071.mjs"), "./locale/lv.js": () => import("./lv.a7031be3.mjs"), "./locale/me.js": () => import("./me.14eab3aa.mjs"), "./locale/mi.js": () => import("./mi.5879a9b9.mjs"), "./locale/mk.js": () => import("./mk.432087fb.mjs"), "./locale/ml.js": () => import("./ml.f7d306ba.mjs"), "./locale/mn.js": () => import("./mn.c3cc33ff.mjs"), "./locale/mr.js": () => import("./mr.20e10ac0.mjs"), "./locale/ms-my.js": () => import("./ms-my.485a95c1.mjs"), "./locale/ms.js": () => import("./ms.b1b34dab.mjs"), "./locale/mt.js": () => import("./mt.2d3d2002.mjs"), "./locale/my.js": () => import("./my.708c03e4.mjs"), "./locale/nb.js": () => import("./nb.23829e5b.mjs"), "./locale/ne.js": () => import("./ne.79456e3e.mjs"), "./locale/nl-be.js": () => import("./nl-be.9cb174ea.mjs"), "./locale/nl.js": () => import("./nl.d729fd27.mjs"), "./locale/nn.js": () => import("./nn.b1b96b18.mjs"), "./locale/oc-lnc.js": () => import("./oc-lnc.93f927f7.mjs"), "./locale/pa-in.js": () => import("./pa-in.15ec6667.mjs"), "./locale/pl.js": () => import("./pl.a699209a.mjs"), "./locale/pt-br.js": () => import("./pt-br.780f4e15.mjs"), "./locale/pt.js": () => import("./pt.0223152c.mjs"), "./locale/rn.js": () => import("./rn.1d0110c4.mjs"), "./locale/ro.js": () => import("./ro.fb8ca3c0.mjs"), "./locale/ru.js": () => import("./ru.6dcf313d.mjs"), "./locale/rw.js": () => import("./rw.36d3ffa4.mjs"), "./locale/sd.js": () => import("./sd.9b1c8e14.mjs"), "./locale/se.js": () => import("./se.3bb9690a.mjs"), "./locale/si.js": () => import("./si.e6bc0e4d.mjs"), "./locale/sk.js": () => import("./sk.95b10b81.mjs"), "./locale/sl.js": () => import("./sl.05cbe403.mjs"), "./locale/sq.js": () => import("./sq.717f53f2.mjs"), "./locale/sr-cyrl.js": () => import("./sr-cyrl.6805b948.mjs"), "./locale/sr.js": () => import("./sr.5ed2e405.mjs"), "./locale/ss.js": () => import("./ss.3394c2dd.mjs"), "./locale/sv-fi.js": () => import("./sv-fi.e0550052.mjs"), "./locale/sv.js": () => import("./sv.f5a50b86.mjs"), "./locale/sw.js": () => import("./sw.f716cf9e.mjs"), "./locale/ta.js": () => import("./ta.37547e53.mjs"), "./locale/te.js": () => import("./te.2496903d.mjs"), "./locale/tet.js": () => import("./tet.f23aa4ce.mjs"), "./locale/tg.js": () => import("./tg.16fd2866.mjs"), "./locale/th.js": () => import("./th.4b206d42.mjs"), "./locale/tk.js": () => import("./tk.4ad1441a.mjs"), "./locale/tl-ph.js": () => import("./tl-ph.29386921.mjs"), "./locale/tlh.js": () => import("./tlh.9565cc50.mjs"), "./locale/tr.js": () => import("./tr.3681f260.mjs"), "./locale/tzl.js": () => import("./tzl.4b85ae24.mjs"), "./locale/tzm-latn.js": () => import("./tzm-latn.78bc985f.mjs"), "./locale/tzm.js": () => import("./tzm.92eb2c1e.mjs"), "./locale/ug-cn.js": () => import("./ug-cn.417ace1e.mjs"), "./locale/uk.js": () => import("./uk.3c82720a.mjs"), "./locale/ur.js": () => import("./ur.5f715131.mjs"), "./locale/uz-latn.js": () => import("./uz-latn.34104044.mjs"), "./locale/uz.js": () => import("./uz.b317a9ef.mjs"), "./locale/vi.js": () => import("./vi.6e2af73c.mjs"), "./locale/x-pseudo.js": () => import("./x-pseudo.23e3be09.mjs"), "./locale/yo.js": () => import("./yo.9654a33f.mjs"), "./locale/zh-cn.js": () => import("./zh-cn.e319c0a5.mjs"), "./locale/zh-hk.js": () => import("./zh-hk.adf6ca00.mjs"), "./locale/zh-tw.js": () => import("./zh-tw.67418955.mjs"), "./locale/zh.js": () => import("./zh.566226b8.mjs") });
        for (const k in h)
          h[k]().then(() => {
            s.locale(d);
            let o, p;
            if (m()) {
              if (O()) {
                if (e.modelValue.length > 0) {
                  const [Y, z] = e.modelValue;
                  o = s(Y, e.formatter.date, !0), p = s(z, e.formatter.date, !0);
                }
              } else if (x()) {
                if (!Pt(e.modelValue))
                  try {
                    console.log(Object.keys(e.modelValue));
                  } catch {
                    console.warn(
                      "[Vue Tailwind Datepicker]: It looks like you want to use Object as the argument %cv-model",
                      "font-style: italic; color: #42b883;",
                      ", but you pass it undefined or null."
                    ), console.warn(
                      "[Vue Tailwind Datepicker]: We has replace with %c{ startDate: '', endDate: '' }",
                      "font-style: italic; color: #42b883;",
                      ", but you can replace manually."
                    ), a("update:modelValue", {
                      startDate: "",
                      endDate: ""
                    });
                  }
                if (e.modelValue) {
                  const [Y, z] = Object.values(e.modelValue);
                  o = Y && s(Y, e.formatter.date, !0), p = z && s(z, e.formatter.date, !0);
                }
              } else if (e.modelValue) {
                const [Y, z] = e.modelValue.split(e.separator);
                o = s(Y, e.formatter.date, !0), p = s(z, e.formatter.date, !0);
              }
              o && p ? (b.value = f(
                {
                  previous: o,
                  next: p
                },
                e
              ), p.isBefore(o, "month") ? (r.value.previous = p, r.value.next = o, r.value.year.previous = p.year(), r.value.year.next = o.year()) : p.isSame(o, "month") ? (r.value.previous = o, r.value.next = p.add(1, "month"), r.value.year.previous = o.year(), r.value.year.next = o.add(1, "year").year()) : (r.value.previous = o, r.value.next = p, r.value.year.previous = o.year(), r.value.year.next = p.year()), e.autoApply || (g.value = [o, p])) : (r.value.previous = s(e.startFrom), r.value.next = s(e.startFrom).add(1, "month"), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year());
            } else {
              if (O()) {
                if (e.modelValue.length > 0) {
                  const [Y] = e.modelValue;
                  o = s(Y, e.formatter.date, !0);
                }
              } else if (x()) {
                if (e.modelValue) {
                  const [Y] = Object.values(e.modelValue);
                  o = s(Y, e.formatter.date, !0);
                }
              } else if (e.modelValue.length) {
                const [Y] = e.modelValue.split(e.separator);
                o = s(Y, e.formatter.date, !0);
              }
              o && o.isValid() ? (b.value = w(o, e), r.value.previous = o, r.value.next = o.add(1, "month"), r.value.year.previous = o.year(), r.value.year.next = o.add(1, "year").year(), e.autoApply || (g.value = [o])) : (r.value.previous = s(e.startFrom), r.value.next = s(e.startFrom).add(1, "month"), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year());
            }
            r.value.weeks = Z() ? J(s.weekdaysShort()) : s.weekdaysShort(), r.value.months = e.formatter.month === "MMM" ? s.monthsShort() : s.months();
          }).catch((o) => {
            console.warn(
              o.message
            );
          });
      });
    });
    const Le = (d) => (d && L.value === null && (L.value = j(y.value)), d && L.value ? "place-right" : "place-left"), Be = (d) => (d && L.value === null && (L.value = j(y.value)), L.value ? "left-auto right-0" : "left-0 right-auto");
    return de("isBetweenRange", re), de("betweenRangeClasses", me), de("datepickerClasses", pe), de("atMouseOver", X), de("setToToday", ve), de("setToYesterday", oe), de("setToLastDay", ie), de("setToThisMonth", ye), de("setToLastMonth", be), de("setToCustomShortcut", Ae), (d, h) => e.noInput ? P.value ? (W(), K("div", so, [
      C("div", lo, [
        C("div", io, [
          e.shortcuts ? (W(), Ie(ot, {
            key: 0,
            shortcuts: e.shortcuts,
            "as-range": m(),
            "as-single": e.asSingle,
            i18n: e.options.shortcuts
          }, null, 8, ["shortcuts", "as-range", "as-single", "i18n"])) : he("", !0),
          C("div", uo, [
            m() && !e.asSingle ? (W(), K("div", co, mo)) : he("", !0),
            C("div", {
              class: Me(["relative w-full sm:w-80", {
                "mb-3 sm:mb-0 sm:mr-2": m() && !e.asSingle
              }])
            }, [
              ee(Ue, {
                panel: l.previous,
                calendar: N(_).previous
              }, null, 8, ["panel", "calendar"]),
              C("div", po, [
                le(ee(ze, {
                  months: N(H),
                  "onUpdate:month": N(_).previous.setMount
                }, null, 8, ["months", "onUpdate:month"]), [
                  [ue, l.previous.month]
                ]),
                le(ee(Ze, {
                  years: N(_).previous.years(),
                  "onUpdate:year": N(_).previous.setYear
                }, null, 8, ["years", "onUpdate:year"]), [
                  [ue, l.previous.year]
                ]),
                le(C("div", null, [
                  ee(We, { weeks: N(M) }, null, 8, ["weeks"]),
                  ee(Ge, {
                    calendar: N(_).previous,
                    weeks: N(M),
                    "as-range": m(),
                    "onUpdate:date": h[2] || (h[2] = (k, o) => R(k, o))
                  }, null, 8, ["calendar", "weeks", "as-range"])
                ], 512), [
                  [ue, l.previous.calendar]
                ])
              ])
            ], 2),
            m() && !e.asSingle ? (W(), K("div", vo, [
              ee(Ue, {
                "as-prev-or-next": "",
                panel: l.next,
                calendar: N(_).next
              }, null, 8, ["panel", "calendar"]),
              C("div", ho, [
                le(ee(ze, {
                  months: N(H),
                  "onUpdate:month": N(_).next.setMount
                }, null, 8, ["months", "onUpdate:month"]), [
                  [ue, l.next.month]
                ]),
                le(ee(Ze, {
                  "as-prev-or-next": "",
                  years: N(_).next.years(),
                  "onUpdate:year": N(_).next.setYear
                }, null, 8, ["years", "onUpdate:year"]), [
                  [ue, l.next.year]
                ]),
                le(C("div", null, [
                  ee(We, { weeks: N(M) }, null, 8, ["weeks"]),
                  ee(Ge, {
                    "as-prev-or-next": "",
                    calendar: N(_).next,
                    weeks: N(M),
                    "as-range": m(),
                    "onUpdate:date": h[3] || (h[3] = (k, o) => R(k, o))
                  }, null, 8, ["calendar", "weeks", "as-range"])
                ], 512), [
                  [ue, l.next.calendar]
                ])
              ])
            ])) : he("", !0)
          ])
        ]),
        e.autoApply ? he("", !0) : (W(), K("div", yo, [
          C("div", bo, [
            C("div", xo, [
              C("button", {
                type: "button",
                class: "away-apply-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-vtd-primary-600 text-base font-medium text-white hover:bg-vtd-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800 disabled:cursor-not-allowed",
                disabled: e.asSingle ? g.value.length < 1 : g.value.length < 2,
                onClick: h[4] || (h[4] = (k) => q()),
                textContent: ne(e.options.footer.apply)
              }, null, 8, go)
            ])
          ])
        ]))
      ])
    ])) : he("", !0) : (W(), Ie(N(yt), {
      key: 0,
      as: "div",
      id: "vtd",
      class: "relative w-full"
    }, {
      default: _e(({ open: k }) => [
        e.overlay ? (W(), Ie(N(Or), {
          key: 0,
          class: "fixed inset-0 bg-black opacity-30"
        })) : he("", !0),
        ee(N(Dr), {
          as: "label",
          class: "relative block"
        }, {
          default: _e(() => [
            Ct(d.$slots, "default", {
              value: b.value,
              placeholder: G.value,
              clear: F
            }, () => [
              le(C("input", Yt({
                ref_key: "VtdInputRef",
                ref: $,
                type: "text",
                class: [
                  "relative block w-full",
                  t.inputClasses || "pl-3 pr-12 py-2.5 rounded-lg overflow-hidden border-solid text-sm text-vtd-secondary-700 placeholder-vtd-secondary-400 transition-colors bg-white border border-vtd-secondary-300 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700 dark:text-vtd-secondary-100 dark:placeholder-vtd-secondary-500 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-20"
                ],
                autocomplete: "off",
                "data-lpignore": "true",
                "data-form-type": "other"
              }, d.$attrs, {
                "onUpdate:modelValue": h[0] || (h[0] = (o) => b.value = o),
                placeholder: G.value,
                onKeyup: A
              }), null, 16, Er), [
                [Tt, b.value]
              ]),
              C("div", Fr, [
                C("button", {
                  type: "button",
                  class: "px-2 py-1 mr-1 focus:outline-none text-vtd-secondary-400 dark:text-opacity-70 rounded-md",
                  onClick: h[1] || (h[1] = (o) => b.value ? F() : d.$refs.VtdInputRef.focus())
                }, [
                  (W(), K("svg", Nr, [
                    b.value ? (W(), K("path", Hr)) : (W(), K("path", Rr))
                  ]))
                ])
              ])
            ])
          ]),
          _: 3
        }),
        ee(nt, {
          "enter-from-class": "opacity-0 translate-y-3",
          "enter-to-class": "opacity-100 translate-y-0",
          "enter-active-class": "transform transition ease-out duration-200",
          "leave-active-class": "transform transition ease-in duration-150",
          "leave-from-class": "opacity-100 translate-y-0",
          "leave-to-class": "opacity-0 translate-y-3"
        }, {
          default: _e(() => [
            ee(N(_r), {
              as: "div",
              class: "relative z-50"
            }, {
              default: _e(({ close: o }) => [
                C("div", {
                  class: Me(["absolute z-50 top-full sm:mt-2.5", Be(k)])
                }, [
                  C("div", {
                    ref_key: "VtdRef",
                    ref: y,
                    class: "fixed inset-0 z-50 overflow-y-auto sm:overflow-visible sm:static sm:z-auto bg-white dark:bg-vtd-secondary-800 sm:rounded-lg shadow-sm"
                  }, [
                    C("div", {
                      class: Me(["vtd-datepicker static sm:relative w-full bg-white sm:rounded-lg sm:shadow-sm border-0 sm:border border-black/[.1] px-3 py-3 sm:px-4 sm:py-4 dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700/[1]", Le(k)])
                    }, [
                      C("div", Ir, [
                        e.shortcuts ? (W(), Ie(ot, {
                          key: 0,
                          shortcuts: e.shortcuts,
                          "as-range": m(),
                          "as-single": e.asSingle,
                          i18n: e.options.shortcuts,
                          close: o
                        }, null, 8, ["shortcuts", "as-range", "as-single", "i18n", "close"])) : he("", !0),
                        C("div", Ur, [
                          m() && !e.asSingle ? (W(), K("div", zr, Zr)) : he("", !0),
                          C("div", {
                            class: Me(["relative w-full sm:w-80", {
                              "mb-3 sm:mb-0 sm:mr-2": m() && !e.asSingle
                            }])
                          }, [
                            ee(Ue, {
                              panel: l.previous,
                              calendar: N(_).previous
                            }, null, 8, ["panel", "calendar"]),
                            C("div", Gr, [
                              le(ee(ze, {
                                months: N(H),
                                "onUpdate:month": N(_).previous.setMount
                              }, null, 8, ["months", "onUpdate:month"]), [
                                [ue, l.previous.month]
                              ]),
                              le(ee(Ze, {
                                years: N(_).previous.years(),
                                "onUpdate:year": N(_).previous.setYear
                              }, null, 8, ["years", "onUpdate:year"]), [
                                [ue, l.previous.year]
                              ]),
                              le(C("div", null, [
                                ee(We, { weeks: N(M) }, null, 8, ["weeks"]),
                                ee(Ge, {
                                  calendar: N(_).previous,
                                  weeks: N(M),
                                  "as-range": m(),
                                  "onUpdate:date": (p, Y) => R(p, Y, o)
                                }, null, 8, ["calendar", "weeks", "as-range", "onUpdate:date"])
                              ], 512), [
                                [ue, l.previous.calendar]
                              ])
                            ])
                          ], 2),
                          m() && !e.asSingle ? (W(), K("div", Kr, [
                            ee(Ue, {
                              "as-prev-or-next": "",
                              panel: l.next,
                              calendar: N(_).next
                            }, null, 8, ["panel", "calendar"]),
                            C("div", Jr, [
                              le(ee(ze, {
                                months: N(H),
                                "onUpdate:month": N(_).next.setMount
                              }, null, 8, ["months", "onUpdate:month"]), [
                                [ue, l.next.month]
                              ]),
                              le(ee(Ze, {
                                "as-prev-or-next": "",
                                years: N(_).next.years(),
                                "onUpdate:year": N(_).next.setYear
                              }, null, 8, ["years", "onUpdate:year"]), [
                                [ue, l.next.year]
                              ]),
                              le(C("div", null, [
                                ee(We, { weeks: N(M) }, null, 8, ["weeks"]),
                                ee(Ge, {
                                  "as-prev-or-next": "",
                                  calendar: N(_).next,
                                  weeks: N(M),
                                  "as-range": m(),
                                  "onUpdate:date": (p, Y) => R(p, Y, o)
                                }, null, 8, ["calendar", "weeks", "as-range", "onUpdate:date"])
                              ], 512), [
                                [ue, l.next.calendar]
                              ])
                            ])
                          ])) : he("", !0)
                        ])
                      ]),
                      e.autoApply ? (W(), K("div", ro, [
                        C("div", oo, [
                          C("div", ao, [
                            C("button", {
                              type: "button",
                              onClick: (p) => o(),
                              class: "away-cancel-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-vtd-secondary-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-vtd-secondary-700 hover:bg-vtd-secondary-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800",
                              textContent: ne(e.options.footer.cancel)
                            }, null, 8, no)
                          ])
                        ])
                      ])) : (W(), K("div", qr, [
                        C("div", Xr, [
                          C("div", Qr, [
                            C("button", {
                              type: "button",
                              class: "away-apply-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-vtd-primary-600 text-base font-medium text-white hover:bg-vtd-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800 disabled:cursor-not-allowed",
                              disabled: e.asSingle ? g.value.length < 1 : g.value.length < 2,
                              onClick: (p) => q(o),
                              textContent: ne(e.options.footer.apply)
                            }, null, 8, eo),
                            C("button", {
                              type: "button",
                              onClick: (p) => o(),
                              class: "mt-3 away-cancel-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-vtd-secondary-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-vtd-secondary-700 hover:bg-vtd-secondary-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800",
                              textContent: ne(e.options.footer.cancel)
                            }, null, 8, to)
                          ])
                        ])
                      ]))
                    ], 2)
                  ], 512)
                ], 2)
              ]),
              _: 2
            }, 1024)
          ]),
          _: 2
        }, 1024)
      ]),
      _: 3
    }));
  }
}, Mt = /* @__PURE__ */ (() => {
  const t = ko;
  return t.install = (a) => {
    a.component("VueTailwindDatepicker", t);
  }, t;
})(), wo = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: Mt
}, Symbol.toStringTag, { value: "Module" }));
Object.entries(wo).forEach(([t, a]) => {
  t !== "default" && (Mt[t] = a);
});
export {
  Mt as c,
  s as d
};
